<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>BWA.bwasw (v2) BETA</title>
    <meta name="slug" content="modules/docs/BWA.bwasw/2" />
    <meta name="template" content="module_doc" />
</head>
<body>
<div class="bs-callout bs-callout-danger">
<h4>This module is currently in beta release.  The module and/or documentation may be incomplete.</h4>
</div>
<p style="font-size: 1.2em; margin-top: 20px;">Aligns long sequences (&amp;gt; 200 bp) to a sequence database using BWA 0.7.4 bwasw.</p>
<div class="row">
<div class="col-sm-4">
<p><strong>Author: </strong>Heng Li, Broad Institute</p>
</div>
<div class="col-sm-4">
<p><strong>Contact: </strong><p>
	gp-help@broadinstitute.org</p>
</p>
</div>
<div class="col-sm-4">
<p><strong>Algorithm Version: </strong>BWA 0.7.4</p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Summary</h2>
<p>
	BWA.bwasw is a fast, light-weight tool that aligns long reads (&gt;200 bp). It performs heuristic Smith-Waterman-like alignment to find high-scoring local hits, and thus chimera. BWA.bwasw can also be used to align ~100bp reads, but it is slower than the BWA.aln module.</p>
<p>
	This document is adapted from the BWA documentation for release 0.7.4.  For more information about BWA.bwasw, see the <a href="http://bio-bwa.sourceforge.net/">BWA project site</a>. BWA.bwasw was developed at the Wellcome Trust Sanger Institute and the Broad Institute.</p>
<p>
<strong style="font-size: 13px; color: rgb(34, 34, 34);">Note: </strong><span style="color: rgb(34, 34, 34);">Index files created with BWA version 0.5.x or earlier are </span><strong style="font-size: 13px; color: rgb(34, 34, 34);">not</strong><span style="color: rgb(34, 34, 34);"> compatible with the aligners of version 0.6.x and newer.  Likewise, the BWA 0.6.x and newer index files are not compatible with the 0.5.x aligners.  The BWA 0.7.x aligners are able to use index files created with 0.6.x, however.</span></p>
<h2>
	Speed</h2>
<p>
	Speed of alignment is largely determined by the error rate of the query sequences, faster with near-perfect hits and slower for higher error rates. Pairing is slower for shorter reads, mostly because shorter reads have more spurious hits.</p>
<h2>
	References</h2>
<p>
	BWA manual page: <a href="http://bio-bwa.sourceforge.net/bwa.shtml">http://bio-bwa.sourceforge.net/bwa.shtml</a>.</p>
<p>
	Li H, Durbin R. Fast and accurate long-read alignment with Burrows-Wheeler transform. <em>Bioinformatics</em>. 2010;26:589-595. [PMID: 20080505] (<a href="http://bioinformatics.oxfordjournals.org/content/26/5/589.long">http://bioinformatics.oxfordjournals.org/content/26/5/589.long</a>)</p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Parameters</h2>
<table id="module_params_table">
<thead>
<tr>
<th>
				Name</th>
<th>
				Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>
				BWA index <span style="color: rgb(255, 0, 0);">*</span></td>
<td>
				A BWA index. You can select from a list of hosted indexes or provide a custom index in the form of a ZIP bundle (as generated by the BWA.indexer module).</td>
</tr>
<tr>
<td>
				read file <span style="color:red;">*</span></td>
<td>
				Single-end or first paired-end reads file in FASTA or FASTQ format.  For paired-end data, this should be the forward ("*_1" or "left") input file.</td>
</tr>
<tr>
<td>
				mate read file <span style="color: rgb(255, 0, 0);">*</span></td>
<td>
<p class="p1">
					The reverse ("*_2" or "right") reads file for paired-end reads in FASTA or FASTQ format.</p>
</td>
</tr>
<tr>
<td>
				match score</td>
<td>
				The score of the match</td>
</tr>
<tr>
<td>
				mismatch penalty</td>
<td>
				Specifies the mismatch penalty.</td>
</tr>
<tr>
<td>
				gap open penalty</td>
<td>
				Gap open penalty.  The gap open penalty is the score taken away for the initiation of the gap in sequence. To make the match more significant you can try to make the gap penalty larger.</td>
</tr>
<tr>
<td>
				gap extension penalty</td>
<td>
				Gap extension penalty.  The gap extension penalty is added to the standard gap penalty for each base or residue in the gap.  To reduce long gaps, increase the extension gap penalty. A few long gaps are expected, rather than many short gaps, so the gap extension penalty should be lower than the gap penalty. (The exception to this rule is where one or both sequences are single reads with possible sequencing errors, in which case many single base gaps are expected. To cope with this, try setting the gap open penalty very low and using the gap extension penalty to control gap scoring.)</td>
</tr>
<tr>
<td>
				band width</td>
<td>
				band width in the banded alignment</td>
</tr>
<tr>
<td>
				min score threshold</td>
<td>
				Minimum score threshold divided by the match score</td>
</tr>
<tr>
<td>
				threshold coefficient</td>
<td>
				Coefficient for threshold adjustment according to query length</td>
</tr>
<tr>
<td>
				z best heuristics</td>
<td>
				Z-best heuristics. Specifying a higher number increases accuracy at the cost of speed</td>
</tr>
<tr>
<td>
				max sa interval size</td>
<td>
				Maximum SA interval size for initiating a seed. Specifying a higher number increases accuracy at the cost of speed.</td>
</tr>
<tr>
<td>
				min num seeds</td>
<td>
				The minimum number of seeds contained in the best alignment from the forward-forward alignment process that allows the algorithm to skip performing the reverse alignment. BWA.bwasw tends to be faster and more accurate if the alignment is supported by more seeds.</td>
</tr>
<tr>
<td>
				num threads</td>
<td>
				Number of threads.  BWA.bwasw will run more quickly with multiple threads; however, there have been reports that results will not always be identical.</td>
</tr>
<tr>
<td>
				output prefix <span style="color:red;">*</span></td>
<td>
				Prefix to use for the output file name</td>
</tr>
</tbody>
</table>
<p>
<span style="color: red;">*</span> - required</p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Input Files</h2>
<ol>
<li style="padding-bottom: 8px;">
		BWA index<br/>
		A set of BWA index files bundled as a ZIP archive, as produced by the BWA.indexer module.  The GenePattern FTP site also hosts a number of index bundels, available in a dropdown selection (requires GenePattern 3.7.0+).<br/>
<strong style="font-size: 13px;">Note:</strong> these index files must have been produced by BWA version 0.6.x or 0.7.x.</li>
<li style="padding-bottom: 8px;">
		read file<br/>
		Single-end or first paired-end reads file in FASTA or FASTQ format.  For paired-end data, this should be the forward ("*_1" or "left") input file.</li>
<li style="padding-bottom: 8px;">
		mate read file<br/>
		The reverse ("*_2" or "right") reads file for paired-end reads in FASTA or FASTQ format.</li>
</ol>
<h2>
	Output Files</h2>
<ol>
<li style="padding-bottom: 8px;">
		SAM file<br/>
		The aligned sequences are output in SAM format. For more details on this alignment file, see the SAM format specification at <a href="http://samtools.sourceforge.net/SAM-1.3.pdf" style="color: rgb(102, 153, 255);">http://samtools.sourceforge.net/SAM-1.3.pdf</a>.</li>
</ol>
</div>
</div>
<h2>Platform Dependencies</h2>
<div class="row">
<div class="col-md-3">
<p><strong>Task Type:</strong><br/>
					RNA-seq</p>
</div>
<div class="col-md-3">
<p><strong>CPU Type:</strong><br/>
					any</p>
</div>
<div class="col-md-3">
<p><strong>Operating System:</strong><br/>
					any</p>
</div>
<div class="col-md-3">
<p><strong>Language:</strong><br/>
					C;Perl</p>
</div>
</div>
<h2>Version Comments</h2>
<table class="table table-striped">
<thead>
<tr>
<th>Version</th>
<th>Release Date</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>2</td>
<td>2014-07-17</td>
<td>Beta: Updated to BWA 0.7.4, changed to use dynamic FTP-hosted index files, and switched to HTML-based doc</td>
</tr>
<tr>
<td>1</td>
<td>2011-05-02</td>
<td></td>
</tr>
</tbody>
</table>
</body>
</html>