<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>SNPFileCreator_SNP6 (v2) BETA</title>
    <meta name="slug" content="modules/docs/SNPFileCreator_SNP6/2" />
    <meta name="template" content="module_doc" />
</head>
<body>
<div class="bs-callout bs-callout-danger">
<h4>This module is currently in beta release.  The module and/or documentation may be incomplete.</h4>
</div>
<p style="font-size: 1.2em; margin-top: 20px;">Performs normalization and probe-level summarization to generate a SNP file for a set of Affymetrix SNP chip CEL files.</p>
<div class="row">
<div class="col-sm-4">
<p><strong>Author: </strong>David Twomey, Joshua Gould, Marc-Danie Nazaire</p>
</div>
<div class="col-sm-4">
<p><strong>Contact: </strong><p>

	gp-help@broadinstitute.org</p>
</p>
</div>
<div class="col-sm-4">
<p><strong>Algorithm Version: </strong></p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>

	Summary</h2>
<div>

	The SNPFileCreator_SNP6 module accepts a ZIP archive containing CEL files generated using an Affymetrix SNP 6.0 chip and, optionally, TXT files with SNP genotype call data. It can be set to perform probe intensity normalization using QuantileNormalization, and probe-level summarization using the Median Polish (default), Average Difference, or Perfect Match/Mismatch (PM/MM) Difference Model (dChip)</div>
<div>

	algorithms. Normalization, summarization, and association with the genotype call data accomplishes the preprocessing usually used for SNP array data.</div>
<div>

	 </div>
<div>

	The module generates a GenePattern SNP file, which contains a matrix of intensity values per probe set and can be used as input for the other SNP 6.0 analysis modules in GenePattern. (Note that SNPFileCreator_SNP6 produces only allele-specific .snp files which are not accepted by some GenePattern SNP modules, intended for use with older chips, such as LOHPaired version 3 and XChromosomeCorrect version 3)</div>
<h2>

	Supported SNP Chips</h2>
<div>

	SNPFileCreator_SNP6 supports CEL files generated using SNP 6.0 chips. It is recommended that you use the GenePattern module, SNPFileCreator v1, to create SNP files for previous version of Affy chips.</div>
<div>

	The module accepts CEL file formats generated by either the Affymetrix® GeneChip® Command Console® Software (AGCC) or Affymetrix GeneChip® Operating Software (GCOS).</div>
<h2>

	Memory Requirement</h2>
<div>

	SNPFileCreator_SNP6 requires on average a minimum of 12GB RAM to process SNP Array 6.0 files. Use the GenePattern public server or configure your local GenePattern server to allocate the required memory for the module. Note that this module is currently only supported on 64-bit Linux servers and is therefore not available for download in the GenePattern Module Repository. </div>
<div>

	 </div>
<div>
<h2>

		References </h2>
</div>
<div>
<ol>
<li>

			Li C, Wong WH. Model-based analysis of oligonucleotide arrays: model validation, design issues and standard error application. Genome Biol. 2001;2:research0032.(<a href="http://genomebiology.com/2001/2/8/research/0032" target="_blank">http://genomebiology.com/2001/2/8/research/0032</a>)</li>
<li>
<div>

				Beroukhim R, Getz G, Nghiemphu L, et al. Assessing the significance of chromosomal aberrations in cancer: methodology and application to glioma. Proc Natl Acad Sci USA. 2007;104:20007-20012. (PDF: <a href="http://www.pnas.org/cgi/reprint/0710052104v1" target="_blank">http://www.pnas.org/cgi/reprint/0710052104v1</a>)</div>
</li>
</ol>
</div>
<p>

	 </p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>

	Parameters</h2>
<table id="module_params_table">
<thead>
<tr>
<th>

				Name</th>
<th>

				Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="width: 150px;">

				cel files <span style="color:red;">*</span></td>
<td>
<p>

					One or more individual 6.0 CEL files or a directory containing 6.0 CEL files.</p>
<div>

					Note: The CEL files must match the Chip Description File (CDF) used; this is particularly important if you upload a custom CDF.</div>
</td>
</tr>
<tr>
<td>

				genome annotation file </td>
<td>

				A genome annotation file. A genome annotation file is provided with SNPFileCreator_SNP6. Options are UCSC hg19 and UCSC hg18. 

				<div>

					You must select one of these files OR upload your own genome annotation file if <em>sort by chromosome and location is set </em>to "Sort".</div>
<div>

					If provideing your own annotation file,  the file must be tab delimited and contain the following three columns: Probe Set ID, Chromosome, and Physical Position. The counting of the chromosome locations in the genome annotations should start with 1 and not 0.</div>
</td>
</tr>
<tr>
<td>

				normalization method <span style="color:red;">*</span></td>
<td>
<p>

					Whether to use a normalization method to compensate for systematic differences across chips:</p>
<ul>
<li>

						Quantile Normalization (dChip) (default). This method takes the pooled distribution of probes on all chips. Then, for each value, it computes the quantile of that value in the distribution of probe intensities and transforms the original value to that quantile's value on the reference chip. This can also reduce noise among replicate measures of the same samples.</li>
<li>

						None</li>
</ul>
<div>

					Note: If you need the Invariant Set Normalization method to reproduce an older analysis, you will need to use SNPFileCreator, available on the Broad public server or in the module repository.</div>
</td>
</tr>
<tr>
<td>

				summarization method <span style="color:red;">*</span></td>
<td>
<div>

					Method used to determine the summarized intensity value for each SNP based on the intensity levels of the probes in each probe set:</div>
<ul>
<li>

						Median Polish (default): an iterative method that determines the significance of the values by examining the data gridded into a two-way table; it is resistant to anomalously high or low values</li>
<li>

						Average Difference: calculates the average difference between the PM probes and the MM probes in a probe set; if there are only PM probes on the array, selecting Average Difference will calculate just the average of the PM probes.</li>
<li>

						PM/MM Difference Model (<a href="https://sites.google.com/site/dchipsoft/" target="_blank">dChip</a>): the method dChip uses that is described by <a href="http://www.pnas.org/content/98/1/31.full" target="_blank">Li and Wong (2001)</a></li>
</ul>
<div>

					NOTE: Median polish summarization is always applied if possible, otherwise average difference summarization, to all probe sets accept for SNP_XXX and CN_XXX probe sets.</div>
</td>
</tr>
<tr>
<td>

				sort by chromosome and location <span style="color:red;">*</span></td>
<td>
<div>

					Whether to sort the SNPs in the output file:</div>
<ul>
<li>

						Do no sort</li>
<li>

						Sort (default); this sorts SNPs by chromosome and physical location. Probe sets that do not have location information are omitted from the output file.</li>
</ul>
<div>

					Tip: If you want to view your SNPs in the Integrative Genomics Viewer (IGV), select Sort, as IGV requires that the SNPs be sorted.</div>
</td>
</tr>
<tr>
<td>

				include RandomGC probesets <span style="color:red;">*</span></td>
<td>
<div>

					Whether to include RandomGC probe sets in the output file. The default (no) is to remove the RandomGC probe sets; this is the recommended setting.</div>
</td>
</tr>
<tr>
<td>

				cdf file </td>
<td>
<p>

					Custom CDF file.(.cdf, .cdf.gz) Leave blank to use default internally provided CDF file.</p>
<div>

					Use a custom Chip Description File (CDF), which describes the design of a chip, providing sequence and type information for the probes representing each of the probe sets on the chip. To use default CDF files provided by SNPFileCreator_SNP6, leave this value blank. </div>
<div>

					 </div>
<div>

					To view the list of CDF files provided by SNPFileCreator_SNP6, in GenePattern, click the module and then select  Properties from the menu. The files are listed in the Current files field of the properties form. </div>
</td>
</tr>
<tr>
<td>

				use mismatch probes <span style="color:red;">*</span></td>
<td>
<p>

					Whether to use mismatch (MM) probes in the summarization step:</p>
<ul>
<li>

						yes</li>
<li>

						no (default)</li>
</ul>
</td>
</tr>
<tr>
<td>

				output prefix <span style="color:red;">*</span></td>
<td>
<div>

					Base name for the output files (required). This file will have the extension .snp.</div>
</td>
</tr>
</tbody>
</table>
<p>
<span style="color: red;">*</span> - required</p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>

	Input Files</h2>
<ol>
<li>

		cel.files<br/>
<div>

			One or more individual 6.0 <a href="http://www.broadinstitute.org/cancer/software/genepattern/gp_guides/file-formats/sections/cel" target="_blank">CEL</a> files, a directory containing 6.0 CEL files, or a ZIP archive containing the CEL files.</div>
</li>
<li>

		Chip Description File (CDF) (optional)

		<div>

			The CDF file describes the design of a chip, providing sequence and type information for the probes representing each of the probe sets on the chip. Examples of Affymetrix CDF files can be found on the <a href="http://www.affymetrix.com/support/technical/libraryfilesmain.affx?hightlight=true&amp;rootCategoryId=" target="_blank">Affymetrix website</a>.</div>
</li>
<li>

		Genome annotation file

		<div>

			This file contains chromosome number, transcription starting and ending sites (direction is from p-arm to q-arm for the human genome), and strand indication (a gene’s sense strand relative to the genome sequence). Either one of the provided genome annotation files must be selected or 4a custom genome annotation file must be uploaded. Alternative annotation files can be found on the <a href="http://www.affymetrix.com/support/technical/libraryfilesmain.affx?hightlight=true&amp;rootCategoryId=" target="_blank">Affymetrix</a><a href="http://www.affymetrix.com/support/technical/annotationfilesmain.affx"> website</a> and other sources, but annotation files can also be made by hand. The annotation only needs to contain three columns: Probe Set ID, Chromosome, and Physical Position.</div>
</li>
</ol>
<h2>

	Output Files</h2>
<ol>
<li>

		stats.txt<br/>
<div>

			 The median, maximum, minimum, and 25th, 50th, and 75th percentile SNP intensity values for each sample</div>
</li>
<li>

		.snp

		<div>

			The SNP file (Allele-Specific: contains raw intensity values for each allele per probe set. The SNP format is three columns per sample: intensity value for allele A, intensity value for allele B, and call); the SNPs in this file will be relative to an initial base of 1.</div>
</li>
</ol>
<h2>

	Example Data</h2>
<div>

	cel files: <a href="ftp://ftp.broadinstitute.org/pub/genepattern/example_files/SNPFileCreator_SNP6/snp6_sample_data.zip">ftp://ftp.broadinstitute.org/pub/genepattern/example_files/SNPFileCreator_SNP6/snp6_sample_data.zip</a></div>
</div>
</div>
<h2>Platform Dependencies</h2>
<div class="row">
<div class="col-md-3">
<p><strong>Task Type:</strong><br/>
					SNP Analysis</p>
</div>
<div class="col-md-3">
<p><strong>CPU Type:</strong><br/>
					64 bit</p>
</div>
<div class="col-md-3">
<p><strong>Operating System:</strong><br/>
					any</p>
</div>
<div class="col-md-3">
<p><strong>Language:</strong><br/>
					Java</p>
</div>
</div>
<h2>Version Comments</h2>
<table class="table table-striped">
<thead>
<tr>
<th>Version</th>
<th>Release Date</th>
<th>Description</th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</body>
</html>