{% extends "base.html" %}
{% block html_lang %}{{ page.lang }}{% endblock %}

{% block title %}{{ SITENAME|striptags }} - {{ page.title|striptags }}{%endblock%}

{% block head %}
  {{ super() }}

  {% import 'translations.html' as translations with context %}
  {% if translations.entry_hreflang(page) %}
    {{ translations.entry_hreflang(page) }}
  {% endif %}
{% endblock %}

{% block content %}
    <div class="gp-content-header fluid">
        <div class="container">
            <h1>{{ page.title }}</h1>
        </div>
    </div>

    <div class="container bs-docs-container">
        <div class="row">
            <div id="content" class="col-md-12">
                <br>
                <p>Click the <img alt="Html" src="/theme/images/html-2e5e581181be3846cf322a91f9c847cd.gif"> or <img alt="Pdf" src="/theme/images/pdf-1e6717ae8311c4dbe7c8c62288f73472.gif"> icons to view a module's documentation.</p>
                <p>Modules in the repository can be <a href="#">installed on a local GenePattern server</a>. Most of these modules are also installed on the <a href="https://cloud.genepattern.org/gp">public GenePattern server</a>.</p>
                <form accept-charset="UTF-8" action="/modules" class="form-inline" method="get">
                    <div style="display:none"><input name="utf8" type="hidden" value="âœ“"></div>
                    <p><strong>View by Category:</strong>
                        <select class="form-control" id="taskType" name="taskType" onchange="change_filter()">
                            <option value=""></option>
                        </select>
                    </p>
                </form>
                <table id="modules" class="table table-striped table-hover table-condensed">
                    <thead>
                        <tr>
                            <th>Name</th>
                            <th>Categories</th>
                        </tr>
                    </thead>
                    <tbody></tbody>
                </table>
            </div>
        </div>
    </div>

    <script type="application/javascript">
        function change_filter() {
            const value = document.getElementById('taskType').value;
            filter_modules(value);
        }

        function filter_modules(value) {
            $('#modules_filter').find('input[type=search]').val(value).trigger('input');
        }

        function build_dropdown(payload) {
            let type_set = new Set();
            for (let m of payload.all_modules)
                for (let c of m.categories)
                    type_set.add(c.toLowerCase());
            type_set = Array.from(type_set).sort();

            const dropdown = document.getElementById('taskType');
            for (let t of type_set) {
                const option = document.createElement('option');
                option.value = t;
                option.text = t;
                dropdown.add(option);
            }
        }

        function doc_icon(url) {
            if (url) return `<a href="${url}"><img alt="Html" src="/theme/images/html-2e5e581181be3846cf322a91f9c847cd.gif"></a>`;
            else return `<img alt="Red x" src="/theme/images/red_x-1d52d35b8fadb4db4817786e096b6c9d.gif">`;
        }

        function build_table(payload) {
            const modules_table = $('#modules');
            const tbody = modules_table.find('tbody');

            for (let m of payload.all_modules)
                tbody.append(`<tr><td><strong>${m.name}</strong><br/>${m.description}<br/>${doc_icon(m.documentation)} Version ${m.version}</td><td style="min-width: 200px; text-transform: capitalize">${m.categories.join(', <br/>')}</td></tr>`);

            modules_table.dataTable({
                "lengthChange" :false,
                "paging": false,
                "filter": true,
                "autoWidth": true,
                "info": true,
                "language": {
                    "search": "Filter Results By: ",
                    "info": "Showing _TOTAL_ of _MAX_ modules"
                }
            });
        }

        function get_filter() {
            const queryString = window.location.search;
            const urlParams = new URLSearchParams(queryString);
            return urlParams.get('taskType');
        }

        async function load_modules() {
            let payload;
            try {
                const response = await fetch("https://cloud.genepattern.org/gp/rest/v1/tasks/all.json");
                if (!response.ok) throw new Error(`HTTP error! Status: ${response.status}`);
                payload = await response.json();
            } catch (error) {
                payload = {
                    "all_modules": [{
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00176:2",
                        "name": "AddFCSEventIndex",
                        "description": "Adds indexes to events in a Flow Cytometry Standard (FCS) data file.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00176%3A2",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00198:2",
                        "name": "AddFCSParameter",
                        "description": "Add parameters and their values to a FCS data file",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00198%3A2",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00179:2",
                        "name": "AddNoiseToFCS",
                        "description": "Add noise to specified parameters in an FCS data file.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00179%3A2",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00346:2",
                        "name": "AffySTExpressionFileCreator",
                        "description": "Creates a GCT file from a set of CEL files from Affymetrix ST arrays.",
                        "version": "2",
                        "documentation": "",
                        "categories": ["preprocess & utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00378:2",
                        "name": "Amaretto",
                        "description": "Discovery of driver genes using epigenomic, genomic and transcriptomic data using module networks with penalized regression.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00378%3A2",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00427:41",
                        "name": "AmpliconSuite",
                        "description": "A multithread-enabled quickstart tool for AmpliconArchitect. Performs all preliminary steps (alignment, CNV calling, seed interval detection) required prior to running AmpliconArchitect. PrepareAA supports hg19 (or GRCh37), hg38 as well as mouse genome mm10 (or GRCm38). PrepareAA can also be invoked to start at intermediate stages of the data preparation process.\n<br>\nAll credits go to: https://github.com/jluebeck/PrepareAA \n<br>\nView licenses here: <a href=\"https://github.com/genepattern/AmpliconSuite#licenses\">https://github.com/genepattern/AmpliconSuite#licenses<\/a>\n<br/>\n<p>\nTo increase your GenePattern disk space quote so you can more easily upload the input bam files, <b><a href=\"https://forms.gle/23tkh98vwkhtCeKM7\">click here to join the AmpliconSuiteUsers group on this server.<\/b> <\/a> <\/p>\n \nTo run multiple samples at once, simply drag and drop multiple samples into the input options. There is no need to check the \"Batch\" box. \n\n<br>\nCurrently using Amplicon Architect version 1.3.4. ",
                        "version": "41",
                        "documentation": "",
                        "categories": ["ecdna"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00429:11",
                        "name": "AmpliconSuiteAggregator",
                        "description": "Aggregates individual AmpliconSuite runs. Takes in zip files (or .tar.gz) and aggregates the results. Results can directly be used to load onto AmpliconRepository. ",
                        "version": "11",
                        "documentation": "",
                        "categories": ["ecdna"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00090:2",
                        "name": "ARACNE",
                        "description": "Runs the ARACNE algorithm for reverse engineering cellular networks",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00090%3A2",
                        "categories": ["Pathway Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00049:2",
                        "name": "AreaChange",
                        "description": "Calculates fraction of area under the spectrum that is attributable to signal (area after noise removal / original area)",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00049%3A2",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00172:9",
                        "name": "AuDIT",
                        "description": "Automated Detection of Inaccurate and Imprecise Transitions in MRM Mass Spectrometry",
                        "version": "9",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00172%3A9",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00251:1",
                        "name": "BedToGtf",
                        "description": "Converts BED files to GFF or GTF format",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00251%3A1",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:8080.genepattern.broad.mit.edu:genepatternmodules:822:2",
                        "name": "Beroukhim.Getz.2007.PNAS.Glioma.GISTIC",
                        "version": "2",
                        "documentation": "",
                        "categories": ["pipeline"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1",
                        "name": "BWA.aln",
                        "description": "A fast and accurate short-read alignment tool that allows for mismatches and gaps. Alignments are output in a SAM format file, which provides Phred-scale quality scores for each alignment.  Uses BWA 0.7.4 aln and samse/sampe.",
                        "version": "2.1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00254%3A2.1",
                        "categories": ["rna-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00256:1.17",
                        "name": "BWA.bwasw",
                        "description": "Aligns long sequences (>200 bp) to a sequence database using BWA 0.7.4 bwasw.",
                        "version": "1.17",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00256%3A1.17",
                        "categories": ["rna-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00253:1.10",
                        "name": "BWA.indexer",
                        "description": "Creates a BWT index from a sequence FASTA file for use with the BWA 0.7.4 aligners.",
                        "version": "1.10",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00253%3A1.10",
                        "categories": ["rna-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00056:2",
                        "name": "CART",
                        "description": "Classification and Regression Tree",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00056%3A2",
                        "categories": ["Prediction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00062:2.2",
                        "name": "CARTXValidation",
                        "description": "Classification and Regression Tree Cross-Validation",
                        "version": "2.2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00062%3A2.2",
                        "categories": ["Prediction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00405:5",
                        "name": "CellFie",
                        "description": "CellFie: Cellular Functions InferencE - An alternative approach to capture the breadth of cellular functions by performing a functional analysis of existing biological networks. Please refer to the documentation for details about the procedure.",
                        "version": "5",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00405%3A5",
                        "categories": ["metabolomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00001:1",
                        "name": "ClassNeighbors",
                        "description": "Select genes that most closely resemble a profile",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00001%3A1",
                        "categories": ["Gene List Selection"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00261:4.1",
                        "name": "ClsFileCreator",
                        "description": "A tool to create a class label (CLS) file.",
                        "version": "4.1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00261%3A4.1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broadinstitute.org:cancer.genome.analysis:00057:1",
                        "name": "CNToXCN",
                        "description": "",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroadinstitute.org%3Acancer.genome.analysis%3A00057%3A1",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00376:0.93",
                        "name": "CoGAPS",
                        "description": "Coordinated Gene Activity in Pattern Sets (CoGAPS) implements a Bayesian MCMC matrix factorization algorithm, GAPS, and links it to gene set statistic methods to infer biological process activity. It can be used to perform sparse matrix factorization on any data, and when this data represents biomolecules, to do gene set analysis.",
                        "version": "0.93",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00376%3A0.93",
                        "categories": ["Tutorial"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00134:2.2.0",
                        "name": "CollapseDataset",
                        "description": "Collapses expression values from multiple input ids that map to a single target gene to a single value on a per-sample basis",
                        "version": "2.2.0",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00134%3A2.2.0",
                        "categories": ["gsea"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00127:3.6",
                        "name": "ComBat",
                        "description": "Batch Correction - Not intended for use with single-cell RNA-seq data.",
                        "version": "3.6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00127%3A3.6",
                        "categories": ["batch correction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00430:6.1",
                        "name": "ComBat_Seq",
                        "description": "ComBat_Seq module. A Batch effect adjustment tool for bulk RNA-seq count data. Improvement upon ComBat,  and uses a negative binomial regression to model batch effects. ComBat-seq takes untransformed, raw count matrix as input. Provides adjusted data which preserves the integer nature of counts, so that the adjusted data are compatible with the assumptions of state-of-the-art differential expression software (e.g. edgeR, DESeq2, which request untransformed count data)\nOriginal Documentation: https://github.com/zhangyuqing/ComBat-seq \nCredits go to Zhang et al, https://doi.org/10.1093/nargab/lqaa078",
                        "version": "6.1",
                        "documentation": "",
                        "categories": ["batch correction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00044:11",
                        "name": "ComparativeMarkerSelection",
                        "description": "Identify differentially expressed genes that can discriminate between distinct classes of samples.",
                        "version": "11",
                        "documentation": "",
                        "categories": ["gene list selection"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00052:3",
                        "name": "CompareSpectra",
                        "description": "Compares two spectra to determine similarity",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00052%3A3",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00283:1",
                        "name": "ConcatenateFilelist",
                        "description": "Concatenate all of the items in the input filelist into a single large file.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00283%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00410:3",
                        "name": "Conos.Cluster",
                        "description": "GenePattern Module which implements the clustering and embedding steps of Conos",
                        "version": "3",
                        "documentation": "",
                        "categories": ["singlecell", "clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00409:3",
                        "name": "Conos.Preprocess",
                        "description": "GenePattern Module which implements the preprocessing and PCA steps of Conos",
                        "version": "3",
                        "documentation": "",
                        "categories": ["singlecell", "clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00423:8",
                        "name": "Conos.VisualizeGeneExpression",
                        "description": "This is a minimalistic implementation of visualization options for Conos. This module is intended to be used after <a href=\"https://cloud.genepattern.org/gp/pages/index.jsf?lsid=urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00410:0.1\">Conos.Cluster<\/a> has been ran.",
                        "version": "8",
                        "documentation": "",
                        "categories": ["single-cell", "singlecell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00030:7.2",
                        "name": "ConsensusClustering",
                        "description": "Resampling-based clustering method ",
                        "version": "7.2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00030%3A7.2",
                        "categories": ["Clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:8080.gpbroad.broadinstitute.org:genepatternmodules:345:1.4.20",
                        "name": "ConstellationMap",
                        "description": "A downstream visualization and analysis tool for gene set enrichment results with interactive web visualizer.",
                        "version": "1.4.20",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3A8080.gpbroad.broadinstitute.org%3Agenepatternmodules%3A345%3A1.4.20",
                        "categories": ["gsea", "pathway analysis", "viewer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00426:1",
                        "name": "Convert.Alevin",
                        "description": "Read alevin output files and convert them to anndata (.h5ad) or loom (.loom) format for downstream analysis. Designed to convert Alevin Velocity quantification into a compatible format for scVelo.",
                        "version": "1",
                        "documentation": "",
                        "categories": ["single-cell", "singlecell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00002:2",
                        "name": "ConvertLineEndings",
                        "description": "Converts line endings to the host operating system's format.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00002%3A2",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00038:2",
                        "name": "ConvertToMAGEML",
                        "description": "Converts a gct, res, or odf dataset file to a MAGE-ML file",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00038%3A2",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00188:2",
                        "name": "ConvertToMAGETAB",
                        "description": "A module to export data from GenePattern in MAGE-TAB format",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00188%3A2",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00085:3",
                        "name": "CopyNumberDivideByNormals",
                        "description": "Determines the copy number of a target SNP",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00085%3A3",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00164:3",
                        "name": "CoxRegression",
                        "description": "Cox regression using clinical data and .cls file",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00164%3A3",
                        "categories": ["Survival Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00343:1",
                        "name": "CreateSymlinks",
                        "description": "Creates symlinks to the input files in the job results directory. This is intended to be a helper module for creating scatter-gather pipelines.  Note that this module is not supported on Windows",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00343%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00174:2",
                        "name": "CsvToFcs",
                        "description": "Converts a Comma separated values (CSV) file to a Flow Cytometry Standard (FCS) file.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00174%3A2",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00335:0.21",
                        "name": "CummeRbund.GeneSetReport",
                        "description": "[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a user-provided list of genes.",
                        "version": "0.21",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00335%3A0.21",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00326:0.30",
                        "name": "CummeRbund.QcReport",
                        "description": "[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots and tables related to QC and Global Stats.",
                        "version": "0.30",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00326%3A0.30",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00342:0.11",
                        "name": "CummeRbund.SelectedConditionsReport",
                        "description": "[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a user-provided list of conditions.",
                        "version": "0.11",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00342%3A0.11",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00336:0.17",
                        "name": "CummeRbund.SelectedGeneReport",
                        "description": "[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a single user-specified gene.",
                        "version": "0.17",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00336%3A0.17",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00097:2",
                        "name": "CytoscapeViewer",
                        "description": "Visualizes a gene network",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00097%3A2",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:8080.gpbroad.broadinstitute.org:genepatternmodules:479:0.19",
                        "name": "DAPPLE",
                        "description": "DAPPLE (Disease Association Protein-Protein Link Evaluator) looks for significant physical connectivity among proteins encoded for by genes in loci associated to disease according to protein-protein interactions reported in the literature.",
                        "version": "0.19",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3A8080.gpbroad.broadinstitute.org%3Agenepatternmodules%3A479%3A0.19",
                        "categories": ["Statistical Methods", "Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00268:1",
                        "name": "DeIdentifyFCS",
                        "description": "DeIdentifyFCS an FCS data file; remove keywords from a list or matching a regular expression; useful to anonymize FCS data files and/or to remove specific (e.g., clinical) information ",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00268%3A1",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00362:3",
                        "name": "DESeq2",
                        "description": "A Differential Expression workflow for RNA-Seq data based on the DESeq2 package from Bioconductor ",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00362%3A3",
                        "categories": ["differential expression"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00045:10",
                        "name": "DifferentialExpressionViewer",
                        "description": "Views differential expression results such as from the ComparativeMarkerSelection module. Formerly known as ComparativeMarkerSelectionViewer.",
                        "version": "10",
                        "documentation": "",
                        "categories": ["javascript"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00393:1.1",
                        "name": "DiSCoVER",
                        "description": "Beta implementation of the DiSCoVER algorithm.",
                        "version": "1.1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00393%3A1.1",
                        "categories": ["RNA-seq", "RNA-Seq", "Statistical Methods", "Prediction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00034:1",
                        "name": "DownloadURL",
                        "description": "Downloads a file from a URL",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00034%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00158:4",
                        "name": "ESPPredictor",
                        "description": "Determine which tryptic peptides will generate the highest signal in ESI-MS",
                        "version": "4",
                        "documentation": "",
                        "categories": ["proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00041:12.5",
                        "name": "ExpressionFileCreator",
                        "description": "Creates a RES or GCT file from a set of Affymetrix CEL files.  For IVT arrays only; use AffySTExpressionFileCreator for ST arrays.",
                        "version": "12.5",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00041%3A12.5",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00248:2",
                        "name": "ExprToGct",
                        "description": "Converts an EXPR file to GCT format",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00248%3A2",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00003:2",
                        "name": "ExtractColumnNames",
                        "description": "Lists the sample descriptors from a .res file.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00003%3A2",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00046:4",
                        "name": "ExtractComparativeMarkerResults",
                        "description": "Creates a derived dataset and feature list file from the results of ComparativeMarkerSelection",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00046%3A4",
                        "categories": ["Gene List Selection"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00182:2",
                        "name": "ExtractFCSDataset",
                        "description": "Extract one or more FCS datasets from an FCS data file",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00182%3A2",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00181:2",
                        "name": "ExtractFCSKeywords",
                        "description": "Extracts keyword(s) value(s) from a Flow Cytometry Standard (FCS) file.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00181%3A2",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00180:2",
                        "name": "ExtractFCSParameters",
                        "description": "Extract specified parameters (dimensions) from an FCS data file.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00180%3A2",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00004:3",
                        "name": "ExtractRowNames",
                        "description": "Extracts the row names from a .res, .gct, or .odf file.",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00004%3A3",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00340:1",
                        "name": "FastQC",
                        "description": "Provides quality control metrics on raw sequence data",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00340%3A1",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00175:3",
                        "name": "FcsToCsv",
                        "description": "Converts a Flow Cytometry Standard (FCS) file to a Comma Separated Values (CSV) file.",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00175%3A3",
                        "categories": ["Data Format Conversion", "Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00005:4",
                        "name": "FeatureSummaryViewer",
                        "description": "Feature Summary Viewer",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00005%3A4",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00274:2",
                        "name": "FileSplitter",
                        "description": "Splits a file into equal-sized chunks, given the number of lines per output file",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00274%3A2",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00280:1",
                        "name": "FilterFilelist",
                        "description": "a helper module which generates a filtered filelist from the input filelist and some filter parameters.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00280%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00159:4.2",
                        "name": "FLAMEPreprocess",
                        "description": "Performs preprocessing of .fcs or .txt flow cytometric data files",
                        "version": "4.2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00159%3A4.2",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00250:5",
                        "name": "Fpkm_trackingToGct",
                        "description": "Takes a Cufflinks FPKM_tracking file and converts it to GCT format.",
                        "version": "5",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00250%3A5",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00152:1",
                        "name": "GctToPcl",
                        "description": "Convert .gct to .pcl",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00152%3A1",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00006:4",
                        "name": "GeneListSignificanceViewer",
                        "description": "Views the results of marker analysis",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00006%3A4",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00007:4",
                        "name": "GeneNeighbors",
                        "description": "Select genes that most closely resemble a continuous profile (e.g., another gene)",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00007%3A4",
                        "categories": ["Gene List Selection"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00042:6",
                        "name": "GEOImporter",
                        "description": "Imports data from the Gene Expression Omnibus (GEO)",
                        "version": "6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00042%3A6",
                        "categories": ["preprocess & utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00125:6.15.30",
                        "name": "GISTIC_2.0",
                        "description": "Genomic Identification of Significant Targets in Cancer (version 2.0.23)",
                        "version": "6.15.30",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00125%3A6.15.30",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00008:0",
                        "name": "GlobalAlignment",
                        "description": "standard global sequence analysis",
                        "version": "0",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00008%3A0",
                        "categories": ["Sequence Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.pipeline:00001:3",
                        "name": "Golub.Slonim.1999.Nature.all.aml.pipeline",
                        "description": "ALL/AML methodology, from Golub and Slonim et al., 1999",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.pipeline%3A00001%3A3",
                        "categories": ["pipeline"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00072:20.4.0",
                        "name": "GSEA",
                        "description": "Gene Set Enrichment Analysis.  <strong>If you are using GSEA on RNA-seq data, please read <a href=\"http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Using_RNA-seq_Datasets_with_GSEA\" target=\"_blank\">these guidelines<\/a>.<\/strong>",
                        "version": "20.4.0",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00072%3A20.4.0",
                        "categories": ["gsea", "pathway analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00073:6",
                        "name": "GSEALeadingEdgeViewer",
                        "description": "Leading Edge Viewer for GSEA results",
                        "version": "6",
                        "documentation": "",
                        "categories": ["gsea"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00317:7.4.0",
                        "name": "GSEAPreranked",
                        "description": "Runs the gene set enrichment analysis against a user-supplied ranked list of genes.",
                        "version": "7.4.0",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00317%3A7.4.0",
                        "categories": ["gsea", "pathway analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00435:6",
                        "name": "Harmony",
                        "description": "Harmony is a batch correction tool. Upon use, this module will produce three scatterplots. One plot will show the data pre-Harmony, and can be found in the 'BeforeHarmonyPlot.png' file. In this plot, the x and y axes represent the 1st and 2nd principal components from PCA dimensional reduction (unless specified otherwise by the user). Another plot will show the data post-Harmony, and can be found in the 'AfterHarmonyPlot.png' file. In this plot, the x and y axes the same dimensions used in the pre-Harmony plot are shown, but adjusted by Harmony. Lastly, the third plot shows the pre-Harmony and post-Harmony plots side by side for comparison.",
                        "version": "6",
                        "documentation": "",
                        "categories": ["Uncategorized"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00032:6",
                        "name": "HeatMapImage",
                        "description": "Creates a heat map from a dataset",
                        "version": "6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00032%3A6",
                        "categories": ["Image Creators"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00010:13.11",
                        "name": "HeatMapViewer",
                        "description": "A configurable heat map viewer that provides users with several options for manipulating and visualizing array-based data",
                        "version": "13.11",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00010%3A13.11",
                        "categories": ["javascript"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00009:6",
                        "name": "HierarchicalClustering",
                        "description": "Hierarchical Clustering",
                        "version": "6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00009%3A6",
                        "categories": ["Clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00071:4",
                        "name": "HierarchicalClusteringImage",
                        "description": "Creates an image of the dendrogram generated from HierarchicalClustering",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00071%3A4",
                        "categories": ["Image Creators"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00031:11.5",
                        "name": "HierarchicalClusteringViewer",
                        "description": "Views results of hierarchical clustering.",
                        "version": "11.5",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00031%3A11.5",
                        "categories": ["javascript"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3",
                        "name": "HISAT2.aligner",
                        "description": "HISAT2 is a fast and sensitive alignment program for mapping next-generation sequencing reads (both DNA and RNA) to a population of human genomes (as well as to a single reference genome).",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00367%3A3",
                        "categories": ["rna-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00368:1.9",
                        "name": "HISAT2.indexer",
                        "description": "Indexes one or more Fasta files for use in the HISAT2.aligner.  ",
                        "version": "1.9",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00368%3A1.9",
                        "categories": ["RNA-Seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00412:4",
                        "name": "HTSeq.Count",
                        "description": "Takes alignment files in SAM/BAM format and a feature file in GTF/GFF format (usually with exon annotation), name sorts the alignment file and calculates for each sample the number of reads mapping to each feature. (HTSeq.count v0.11.2)",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00412%3A4",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00011:1",
                        "name": "Hu68kHu35kAtoU95",
                        "description": "Converts a list of Affymetrix Hu6800/Hu35KsubA probes to the corresponding Affymetrix U95Av2 probes.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00011%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5",
                        "name": "igv.js",
                        "description": "Launches igv.js with input file(s)",
                        "version": "2.5",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00363%3A2.5",
                        "categories": ["viewer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00189:2",
                        "name": "IlluminaExpressionFileCreator",
                        "description": "Creates a GCT file from a zip of Illumina IDAT files and an Illumina manifest file",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00189%3A2",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00054:13",
                        "name": "ImputeMissingValues.KNN",
                        "description": "Imputes missing values",
                        "version": "13",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00054%3A13",
                        "categories": ["Missing Value Imputation"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00063:2",
                        "name": "JavaTreeView",
                        "description": "Hierarchical clustering viewer that reads in Eisen's cdt, atr, and gtr files. A detailed description of these formats is available from http://microarray.ccgb.umn.edu/smd/html/MicroArray/help/formats.shtml",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00063%3A2",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00434:2",
                        "name": "Kallisto.Indexer",
                        "description": "Builds an index from a FASTA formatted file of target sequences for use with Kallisto.  ",
                        "version": "2",
                        "documentation": "",
                        "categories": ["rna-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00383:4",
                        "name": "Kallisto.Quant",
                        "description": "Runs Kallisto to perform pseudoalignment (using a human or mouse transcriptome index) and gene-level aggregation",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00383%3A4",
                        "categories": ["rna-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00081:2",
                        "name": "KMeansClustering",
                        "description": "Module that performs the K-Means Clustering algorithm",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00081%3A2",
                        "categories": ["Clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00012:4",
                        "name": "KNN",
                        "description": "K-nearest neighbors classification",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00012%3A4",
                        "categories": ["Prediction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00013:6",
                        "name": "KNNXValidation",
                        "description": "KNN classification with Leave-One-Out Cross-Validation",
                        "version": "6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00013%3A6",
                        "categories": ["Prediction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00014:1",
                        "name": "KSscore",
                        "description": "Computes the Kolmogorov-Smirnov score for a set of genes within an ordered list.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00014%3A1",
                        "categories": ["Statistical Methods"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00083:2",
                        "name": "LandmarkMatch",
                        "description": "A proteomics method to propogate identifed peptides across multiple MS runs",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00083%3A2",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00275:0.3",
                        "name": "ListFiles",
                        "description": "a helper module which outputs a filelist, the list of all files in the input directory, similar to the unix 'ls' command",
                        "version": "0.3",
                        "documentation": "",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00051:2",
                        "name": "LocatePeaks",
                        "description": "Locates detected peaks in a spectrum.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00051%3A2",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00192:1",
                        "name": "LogicleTransformFCS",
                        "description": "Logicle transformation of (selected) parameters in a list mode FCS data file.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00192%3A1",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00094:3",
                        "name": "LOHPaired",
                        "description": "Computes LOH for paired samples",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00094%3A3",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.pipeline:00004:1",
                        "name": "Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline",
                        "description": "Normal/tumor classifier and kNN prediction of mouse lung samples",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.pipeline%3A00004%3A1",
                        "categories": ["pipeline"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00055:2",
                        "name": "MAGEMLImportViewer",
                        "description": "A visualizer to import data in MAGE-ML format into GenePattern",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00055%3A2",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00173:2",
                        "name": "MAGETABImportViewer",
                        "description": "A visualizer to import data in MAGE-TAB format into GenePattern",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00173%3A2",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00015:3",
                        "name": "MapChipFeaturesGeneral",
                        "description": "Change (map) the features (genes) of a dataset",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00015%3A3",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00070:1",
                        "name": "MergeColumns",
                        "description": "Merge datasets by column.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00070%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00187:1",
                        "name": "MergeFCSDataFiles",
                        "description": "Merge multiple FCS data files into a single FCS datasets; includes multimple subsampling options.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00187%3A1",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00354:1",
                        "name": "MergeHTSeqCounts",
                        "description": "Merges HTSeq read count data files into one file",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00354%3A1",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00064:1",
                        "name": "MergeRows",
                        "description": "Merge datasets by row.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00064%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00092:1",
                        "name": "MetageneProjection",
                        "description": "Projects one or more data sets onto the metagene representation of another data set",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00092%3A1",
                        "categories": ["Projection"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00379:4",
                        "name": "MethylationCNVAnalysis",
                        "description": "Methylation array preprocessing with the minfi R package and copy-number variation analysis with the conumee R package. Compatible with both 450k or EPIC array data.",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00379%3A4",
                        "categories": ["Methylation", "CNV Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00107:2",
                        "name": "MINDY",
                        "description": "Runs the MINDY algorithm for inferring genes that modulate the activity of a transcription factor at post-transcriptional levels ",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00107%3A2",
                        "categories": ["Pathway Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00110:1",
                        "name": "MINDYViewer",
                        "description": "Visualizes the results of MINDY",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00110%3A1",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:127.0.0.1:genepatternmodules:4:1.5.62",
                        "name": "MultiplotStudio",
                        "description": "Scatterplot visualization tool for analyzing differential gene expression.",
                        "version": "1.5.62",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3A127.0.0.1%3Agenepatternmodules%3A4%3A1.5.62",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00380:2",
                        "name": "MutPanning",
                        "description": "MutPanning identifies driver genes based on their abundance of nonsynonymous mutations and their increased number of mutations in unusual nucleotide contexts that deviate from the background mutational process. ",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00380%3A2",
                        "categories": ["Mutational Significance Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00320:1.3.5",
                        "name": "MutSigCV",
                        "description": "Identifies genes that are significantly mutated in cancer genomes, using a model with mutational covariates.",
                        "version": "1.3.5",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00320%3A1.3.5",
                        "categories": ["Mutational Significance Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00069:1",
                        "name": "mzXMLToCsv",
                        "description": "Converts a mzXML file to a zip of csv files",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00069%3A1",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00166:6",
                        "name": "NearestTemplatePrediction",
                        "description": "Nearest neighbor prediction based on a list of marker genes",
                        "version": "6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00166%3A6",
                        "categories": ["Prediction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00016:2.1",
                        "name": "NMF",
                        "description": "Non-negative Matrix Factorization",
                        "version": "2.1",
                        "documentation": "",
                        "categories": ["Dimension Reduction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:8080.genepattern.genepattern.sdsc.edu:genepatternmodules:10:7",
                        "name": "NMFClustering",
                        "description": "GPU-backed implementation of non-negative Matrix Factorization (NMF) Consensus Clustering.",
                        "version": "7",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3A8080.genepattern.genepattern.sdsc.edu%3Agenepatternmodules%3A10%3A7",
                        "categories": ["clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00057:5",
                        "name": "NMFConsensus",
                        "description": "Non-negative Matrix Factorization (NMF) Consensus Clustering",
                        "version": "5",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00057%3A5",
                        "categories": ["Clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00411:1.3",
                        "name": "OpenCRAVAT",
                        "description": "<img src=\"https://opencravat.org/images/OpenCRAVAT_Logo-0-sm.png\" width=50px alt=\"img\"/>OpenCRAVAT performs genomic variant interpretation including variant impact, annotation, and scoring.",
                        "version": "1.3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00411%3A1.3",
                        "categories": ["variant annotation"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00431:3",
                        "name": "OutSplice",
                        "description": "OutSplice is a package that compares alternative splicing events between tumor and normal samples. This package is specifically designed for analyzing the gene and junction information from RNA sequencing data provided by the user, or from the TCGA. This package generates a matrix of splicing outliers, which are junctions that are either significantly over or under-expressed compared to a control sample. OutSplice further designates observed outliers as skipping, insertion, or deletion events. Overall, OutSplice is novel in that it determines differential splicing burdens between tumors and normal samples and characterizes the nature of splicing outliers.",
                        "version": "3",
                        "documentation": "",
                        "categories": ["alternative splicing"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00433:3",
                        "name": "OutSplice.plotJunctions",
                        "description": "Using the result file from OutSplice, OutSplice.plotJunctions creates bar and waterfall plots of junction expression in both the tumor and normal samples.",
                        "version": "3",
                        "documentation": "",
                        "categories": ["alternative splicing"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00017:4.2",
                        "name": "PCA",
                        "description": "Principal Component Analysis (adapted from MeV, <a href=\"http://mev.tm4.org\">http://mev.tm4.org<\/a>)",
                        "version": "4.2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00017%3A4.2",
                        "categories": ["Dimension Reduction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00018:6",
                        "name": "PCAViewer",
                        "description": "Principal Component Analysis Visualizer (adapted from MeV, http://www.tigr.org/software/tm4/mev.html)",
                        "version": "6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00018%3A6",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00151:1",
                        "name": "PclToGct",
                        "description": "Convert Stanford preclustering file (.pcl) to .gct",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00151%3A1",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00050:2",
                        "name": "Peaks",
                        "description": "Determine peaks in the spectrum using a series of digital filters.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00050%3A2",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00266:3",
                        "name": "Picard.AddOrReplaceReadGroups",
                        "description": "Replaces all read groups in the input file with a new read group and assigns all reads to this read group in the output",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00266%3A3",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00252:2",
                        "name": "Picard.BamToSam",
                        "description": "Converts a BAM file to a SAM file",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00252%3A2",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00244:1",
                        "name": "Picard.CreateSequenceDictionary",
                        "description": "Reads FASTA or FASTA.GZ files containing reference sequences, and writes them as a SAM file containing a sequence dictionary",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00244%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00241:2",
                        "name": "Picard.FastqToSam",
                        "description": "Extracts read sequences and qualities from the input fastq file and writes them into the output file in unaligned SAM or BAM format",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00241%3A2",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00267:2",
                        "name": "Picard.MarkDuplicates",
                        "description": "Examines aligned records in the supplied SAM or BAM file to locate duplicate reads.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00267%3A2",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00264:1",
                        "name": "Picard.ReorderSam",
                        "description": "Reorders a SAM file or a BAM file to match contig ordering in a provided reference file",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00264%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00234:2",
                        "name": "Picard.SamToBam",
                        "description": "Converts a SAM file to a BAM file",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00234%3A2",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00240:3",
                        "name": "Picard.SamToFastq",
                        "description": "Convert SAM or BAM files to FASTQ format",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00240%3A3",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00245:4",
                        "name": "Picard.SortSam",
                        "description": "Sorts a SAM or BAM file in a specified order",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00245%3A4",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00053:2",
                        "name": "PlotPeaks",
                        "description": "Plot peaks",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00053%3A2",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00019:4.11",
                        "name": "PredictionResultsViewer",
                        "description": "Visualizes prediction results in JavaScript",
                        "version": "4.11",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00019%3A4.11",
                        "categories": ["javascript"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00020:5.1",
                        "name": "PreprocessDataset",
                        "description": "Performs several preprocessing steps on a res, gct, or odf input file",
                        "version": "5.1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00020%3A5.1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00424:1",
                        "name": "PreprocessVelocityTranscriptome",
                        "description": "Extract transcript and intron sequences from the genome sequence using the eisaR package in order to quantify both intronic (unprocessed) and exonic (processed) RNA abundances in each cell with alevin.<br>A necessary preprocessing step to enable RNA velocity analysis.",
                        "version": "1",
                        "documentation": "",
                        "categories": ["preprocess & utilities", "single-cell", "singlecell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00185:2",
                        "name": "PreviewFCS",
                        "description": "Allows viewing of structural metadata, parameters, and descriptive statistics from a Flow Cytometry Standard (FCS) data file",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00185%3A2",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00068:1",
                        "name": "ProteoArray",
                        "description": "LC-MS proteomic data processing module",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00068%3A1",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00048:2",
                        "name": "ProteomicsAnalysis",
                        "description": "Runs the proteomics analysis on the set of input spectra",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00048%3A2",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:8080.gpserver.ip-172-31-26-71.ip-172-31-26-71.ec2.internal:genepatternmodules:69:4",
                        "name": "PTM_SEA",
                        "description": "PTM-SEA for use with PTMsigDB",
                        "version": "4",
                        "documentation": "",
                        "categories": ["Proteomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00417:4",
                        "name": "PyCoGAPS",
                        "description": "Coordinated Gene Activity in Pattern Sets (CoGAPS) implements a Bayesian MCMC matrix factorization algorithm, GAPS, and links it to gene set statistic methods to infer biological process activity. It can be used to perform sparse matrix factorization on any data, and when this data represents biomolecules, to do gene set analysis. This package, PyCoGAPS, presents a unified Python interface, with a parallel, efficient underlying implementation.",
                        "version": "4",
                        "documentation": "",
                        "categories": ["dimension reduction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00436:4",
                        "name": "RandomForest",
                        "description": "Performs random forest classification on data and class files (cross-validation or test-train prediction), outputting a prediction results file.",
                        "version": "4",
                        "documentation": "",
                        "categories": ["prediction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00315:1.10",
                        "name": "RankNormalize",
                        "description": "Normalize the rows in a data set by rank",
                        "version": "1.10",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00315%3A1.10",
                        "categories": ["Statistical Methods"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00339:0.15",
                        "name": "Read_group_trackingToGct",
                        "description": "[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Converts a Cufflinks read_group_tracking file into GCT format",
                        "version": "0.15",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00339%3A0.15",
                        "categories": ["RNA-seq", "Data Format Conversion", "Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00184:1",
                        "name": "RemoveMissingValues",
                        "description": "Removes features and/or samples containing missing (NA) values",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00184%3A1",
                        "categories": ["Missing Value Imputation"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00210:1",
                        "name": "RemoveSaturatedFCSEvents",
                        "description": "Remove saturated events from an FCS data file",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00210%3A1",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1",
                        "name": "RenameFile",
                        "description": "Creates a result file with the contents of the input but with a new name.  Where possible, this is done without copying the file but instead by simply making a link.  While it is possible to run this with a submitted file, the common use will be to change the name of a result file or a previously uploaded file.  Note that the original file will still be present in its original location. ",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00338%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00033:3",
                        "name": "ReorderByClass",
                        "description": "Reorder the samples in an expression dataset and class file by class",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00033%3A3",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15",
                        "name": "RNASeQC",
                        "description": "[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Calculates metrics on aligned RNA-seq data.",
                        "version": "2.15",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00263%3A2.15",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:7070.genepatt.genepatterntest.broadinstitute.org:genepatternmodules:268:6",
                        "name": "SageKda2Cys",
                        "description": "Takes a zip of KeyDriverAnalysis output files and formats them as a cytoscape .cys session file",
                        "version": "6",
                        "documentation": "",
                        "categories": ["Sage"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00418:0.6",
                        "name": "Salmon.Alevin.Quant",
                        "description": "Quantification and analysis of 3' tagged-end single-cell sequencing data using Salmon's Alevin (v1.5.2) function.<br>\n\nSee: <a href=\"https://salmon.readthedocs.io/en/latest/alevin.html\" target=\"_blank\">The Alevin Documentation<\/a>\nfor detailed usage guidelines.",
                        "version": "0.6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Agenepattern.org%3Amodule.analysis%3A00418%3A0.6",
                        "categories": ["single-cell", "rna-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00419:0.5",
                        "name": "Salmon.Indexer",
                        "description": "Salmon (version 1.6.0) transcriptome indexer.<br>\n\nSee: <a href=\"https://salmon.readthedocs.io/en/latest/salmon.html#using-salmon\" target=\"_blank\">The Salmon User Guide<\/a>\nfor detailed usage guidelines.",
                        "version": "0.5",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Agenepattern.org%3Amodule.analysis%3A00419%3A0.5",
                        "categories": ["preprocess & utilities", "rna-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00420:0.9",
                        "name": "Salmon.Quant",
                        "description": "Perform transcript-level quantification of RNA-seq data using Salmon (version 1.6.0).<br>\n\nSee: <a href=\"https://salmon.readthedocs.io/en/latest/salmon.html#using-salmon\" target=\"_blank\">The Salmon User Guide<\/a>\nfor detailed usage guidelines.",
                        "version": "0.9",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Agenepattern.org%3Amodule.analysis%3A00420%3A0.9",
                        "categories": ["rna-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00265:0.3",
                        "name": "SamTools.FastaIndex",
                        "description": "Indexes a reference sequence in the FASTA format",
                        "version": "0.3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00265%3A0.3",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00428:2",
                        "name": "ScanpyQC",
                        "description": "Create Scanpy Quality Control Plots to enable unblinded dataset filtering with ScanpyUtilities ",
                        "version": "2",
                        "documentation": "",
                        "categories": ["singlecell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00382:1.3",
                        "name": "ScanpyUtilities",
                        "description": "Scanpy preprocessing of an h5 single cell RNASeq dataset",
                        "version": "1.3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00382%3A1.3",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00425:2",
                        "name": "scVelo",
                        "description": "Module for performing basic velocity analysis using the scVelo protocol on ScanpyUtilities processed datasets",
                        "version": "2",
                        "documentation": "",
                        "categories": ["single-cell", "rna velocity", "singlecell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00021:4",
                        "name": "SelectFeaturesColumns",
                        "description": "Takes a 'column slice' from a .res, .gct, .snp, .cn, .loh, .odf, or .cls file.",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00021%3A4",
                        "categories": ["Gene List Selection"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00022:4",
                        "name": "SelectFeaturesRows",
                        "description": "Generate a .res, .gct, .snp, .cn, .loh, or .odf file containing only specified features (i.e. rows).",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00022%3A4",
                        "categories": ["Gene List Selection"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1",
                        "name": "SelectFileMatrix",
                        "description": "A helper module which selects a matrix out of a delimited text file.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00285%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broadinstitute.org:cancer.genome.analysis:00065:0.4",
                        "name": "SelectSNPProbes",
                        "description": "Filter to keep just copy number probes",
                        "version": "0.4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroadinstitute.org%3Acancer.genome.analysis%3A00065%3A0.4",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00186:2",
                        "name": "SetFCSKeywords",
                        "description": "Sets keyword/value(s) in a Flow Cytometry Standard (FCS) file.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00186%3A2",
                        "categories": ["Flow Cytometry"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00408:1.2",
                        "name": "Seurat.Clustering",
                        "description": "Companion module for the <a href=\"https://notebook.genepattern.org/services/sharing/notebooks/376/preview/\">Seurat notebook<\/a>. Performs clustering and marker identification on single cell data.",
                        "version": "1.2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00408%3A1.2",
                        "categories": ["singlecell", "clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:8080.gpserver.ip-172-31-26-71.ip-172-31-26-71.ec2.internal:genepatternmodules:941:1.3",
                        "name": "Seurat.IntegrateData",
                        "description": "Batch Correction for Seurat Objects",
                        "version": "1.3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3A8080.gpserver.ip-172-31-26-71.ip-172-31-26-71.ec2.internal%3Agenepatternmodules%3A941%3A1.3",
                        "categories": ["preprocess & utilities", "dimension reduction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00415:4.1",
                        "name": "Seurat.Preprocessing",
                        "description": "Seurat preprocessing",
                        "version": "4.1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Agenepattern.org%3Amodule.analysis%3A00415%3A4.1",
                        "categories": ["preprocess & utilities", "single-cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00416:4.11",
                        "name": "Seurat.QC",
                        "description": "Seurat quality control (First step in the Seurat Suite), using Seurat version 4.0.3.",
                        "version": "4.11",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Agenepattern.org%3Amodule.analysis%3A00416%3A4.11",
                        "categories": ["preprocess & utilities", "single-cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00421:2",
                        "name": "Seurat.VisualizeMarkerExpression",
                        "description": "Visualize Marker Expression as violin plots and on a UMap. To be used after Seurat.Clustering",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Agenepattern.org%3Amodule.analysis%3A00421%3A2",
                        "categories": ["single-cell", "singlecell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00217:0.4",
                        "name": "shRNAscores",
                        "description": "Outputs fold change, signal to noise and modified score (which deweights highly variable hps from reference control) for RNAI HP scoring",
                        "version": "0.4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00217%3A0.4",
                        "categories": ["RNAi"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00437:4",
                        "name": "Single.Cell.ssGSEA",
                        "description": "Perform Single Cell ssGSEA. Can be used after Seurat.Clustering module.",
                        "version": "4",
                        "documentation": "",
                        "categories": ["singlecell", "gsea", "single-cell", "projection", "pathway analysis", "single cell gsea", "singlecell gsea"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00086:3.12",
                        "name": "SNPFileCreator",
                        "description": "Performs normalization and probe-level summarization to generate a SNP file for a set of Affymetrix SNP chip CEL files.",
                        "version": "3.12",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00086%3A3.12",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00095:0.3",
                        "name": "SNPFileCreator_1chip",
                        "description": "Processes Affymetrix SNP probe-level data into expression values",
                        "version": "0.3",
                        "documentation": "",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00302:1.2",
                        "name": "SNPFileCreator_SNP6",
                        "description": "Performs normalization and probe-level summarization to generate a SNP file for a set of Affymetrix SNP chip CEL files.",
                        "version": "1.2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00302%3A1.2",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00096:1",
                        "name": "SNPFileSorter",
                        "description": "Sorts a .snp file by Chromosome and location",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00096%3A1",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00106:1.3",
                        "name": "SNPMultipleSampleAnalysis",
                        "description": "Determine Regions of Concordant Copy Number Aberrations",
                        "version": "1.3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00106%3A1.3",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00084:3",
                        "name": "SnpViewer",
                        "description": "Displays a heat map of snp data",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00084%3A3",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00029:2",
                        "name": "SOMClustering",
                        "description": "Self-Organizing Maps algorithm",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00029%3A2",
                        "categories": ["Clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00023:5.1",
                        "name": "SOMClusterViewer",
                        "description": "Visualize clusters created with the SOM algorithm",
                        "version": "5.1",
                        "documentation": "",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00227:1.1",
                        "name": "SparseHierarchicalClustering",
                        "description": "Cluster the samples in a data matrix using an adaptively-chosen subset of the features.",
                        "version": "1.1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00227%3A1.1",
                        "categories": ["Clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00460:7",
                        "name": "spatialGE.Preprocessing",
                        "description": "The package spatialGE provides a collection of tools for the visualization of gene expression from spatially-resolved transcriptomic experiments. The data input methods have been designed so that any data can be analyzed as long as it contains gene expression counts per region of interest (ROI), spot, or cell, and the spatial coordinates of those ROIs, spots, or cells, as it is generated in platforms such as GeoMx, Visium, and CosMx-SMI. The spatialge.Preprocessing module performs initial data ingestion, filtering, transform and pseudo-bulk operations to prepare data for further processing. Order of operations are: ingestion, filter, pseudobulk, transform.",
                        "version": "7",
                        "documentation": "",
                        "categories": ["spatial transcriptomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00466:1",
                        "name": "spatialGE.SpatialAutocorrelation",
                        "description": "This function performs spatial statistics analysis on spatial transcriptomics data using the SThet module.",
                        "version": "1",
                        "documentation": "",
                        "categories": ["spatial transcriptomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00462:2",
                        "name": "spatialGE.STclust",
                        "description": "This function performs spatial clustering on gene expression data using the STclust method.",
                        "version": "2",
                        "documentation": "",
                        "categories": ["clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00473:1",
                        "name": "spatialGE.STdiff",
                        "description": "STdiff function for differential expression analysis in spatial transcriptomics data.",
                        "version": "1",
                        "documentation": "",
                        "categories": ["differential expression"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00461:3",
                        "name": "spatialGE.STenrich",
                        "description": "Detect genes showing spatial expression patterns (e.g., hotspots). Tests if spots/cells with high average expression of a gene set shows evidence of spatial aggregation.",
                        "version": "3",
                        "documentation": "",
                        "categories": ["spatial transcriptomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00465:2",
                        "name": "spatialGE.STgradient",
                        "description": "Tests for genes for which there is evidence of expression spatial gradients with respect to a \"reference\" tissue niche/domain (e.g., higher expression closer to reference tissue niche, lower expression as farther from reference tissue niche).<br/><br/>If you just ran <strong>spatialGE.STclust<\/strong>, you probably want the <strong>spatialGE.STgradient.Launcher<\/strong> module instead.",
                        "version": "2",
                        "documentation": "",
                        "categories": ["spatial transcriptomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00468:2",
                        "name": "spatialGE.STgradient.Launcher",
                        "description": "Reads the output of spatialGE.STclust and then launches an analysis to test for genes for which there is evidence of expression spatial gradients with respect to a \"reference\" tissue niche/domain (e.g., higher expression closer to reference tissue niche, lower expression as farther from reference tissue niche).",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Agenepattern.org%3Amodule.analysis%3A00468%3A2",
                        "categories": ["spatial transcriptomics, javascript"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00469:1",
                        "name": "spatialGE.STplot",
                        "description": "Plots of gene expression, cluster memberships, and metadata in spatial context",
                        "version": "1",
                        "documentation": "",
                        "categories": ["spatial transcriptomics"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00471:1",
                        "name": "spatialGE.STplot.Launcher",
                        "description": "Generates high-quality figures showing the expression of genes in spatial context. ",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Agenepattern.org%3Amodule.analysis%3A00471%3A1",
                        "categories": ["spatial transcriptomics, javascript"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00300:1",
                        "name": "SplitColumns",
                        "description": "Creates a separate file for each column specified from the input file",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00300%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00024:4",
                        "name": "SplitDatasetTrainTest",
                        "description": "Splits a dataset (and cls file) into a number of train and test subsets",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00024%3A4",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00270:10.1.0",
                        "name": "ssGSEA",
                        "description": "Performs single sample GSEA.  NOTE: with the release of v10.0.1, this module was renamed from \"ssGSEAProjection\" to just \"ssGSEA\". For single-cell data, use the <a href=\"https://genepattern.github.io/Single.Cell.ssGSEA/documentation\">Single.Cell.ssGSEA<\/a> module instead.",
                        "version": "10.1.0",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00270%3A10.1.0",
                        "categories": ["gsea", "pathway analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00414:1",
                        "name": "ssGSEA.ROC",
                        "description": "Generate ROC curves and calculate the AUC from ssGSEA results and a binary phenotype classification.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00414%3A1",
                        "categories": ["gsea", "statistical methods"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00370:2",
                        "name": "STAR.aligner",
                        "description": "STAR (Spliced Transcripts Alignment to a Reference) 2.7.10b is a fast RNA-seq reads to genome mapper. It differs from other mappers as TopHat in that it gains speed at the expense of consuming more RAM and that it incorporates transcriptome annotation at the indexing build stage rather than at the analysis stage. It can optionally detect non-canonical splices and chimeric transcripts.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00370%3A2",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00371:1",
                        "name": "STAR.indexer",
                        "description": "The STAR indexer (version 2.7.10b) makes from a series of fastA files containing a genome an index for use with the STAR spliced read to genome aligner. It can optionally (but highly recommended !) incorporate information about known intron-exon junctions.",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00371%3A1",
                        "categories": ["RNA-seq"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00402:0.12",
                        "name": "STREAM.DetectDifferentiallyExpressedGenes",
                        "description": "Used to detect differentially expressed genes between pairs of branches.",
                        "version": "0.12",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00402%3A0.12",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00403:0.12",
                        "name": "STREAM.DetectLeafGenes",
                        "description": "Detect marker genes for each leaf branch.",
                        "version": "0.12",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00403%3A0.12",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00404:0.3",
                        "name": "STREAM.DetectTransitionGenes",
                        "description": "Detects the transition genes for each branch of a STREAM trajectory.",
                        "version": "0.3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00404%3A0.3",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00397:0.7",
                        "name": "STREAM.DimensionReduction",
                        "description": "Perform dimension reduction on the dataset to make it easier to reliably assess similarity or distances between cells. Starting from the genes selected in the STREAM.FeatureSelection module, we project cells to a lower dimensional space using a non-linear dimensionality reduction method.",
                        "version": "0.7",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00397%3A0.7",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00399:0.4",
                        "name": "STREAM.ElasticPrincipalGraph",
                        "description": "STREAM (Single-cell Trajectories Reconstruction, Exploration And Mapping) is an interactive pipeline capable of disentangling and visualizing complex branching trajectories from both single-cell transcriptomic and epigenomic data. STREAM.ElasticPrincipalGraph is uses elastic principal graph learning to calculate a pseudotime trajectory.",
                        "version": "0.4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00399%3A0.4",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00400:0.6",
                        "name": "STREAM.EPGAdjustFinalGraph",
                        "description": "Adjust the final EPG trajectory structure by pruning, extending and otherwise modifying the branches to optimize for single-cell data",
                        "version": "0.6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00400%3A0.6",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00396:0.6",
                        "name": "STREAM.FeatureSelection",
                        "description": "Select either or both of the most variable genes and then find the principal components using PCA. These will then be available for the downstram trajectory analysis.  ",
                        "version": "0.6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00396%3A0.6",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00406:0.15",
                        "name": "STREAM.Plot2DVisualization",
                        "description": "STREAM.Plot2DVisualization is used to check if there is clear meaningful trajectory pattern via TSNE or UMAP 2-D plots. If so, we can continue the STREAM analysis; If not, go back to previous steps (preprocessing, feature selection and dimension reduction) to modify the parameters.",
                        "version": "0.15",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00406%3A0.15",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00395:0.4",
                        "name": "STREAM.Preprocess",
                        "description": "Preprocess single cell RNA-seq data to prepare it for the STREAM trajectory analysis. This module will normalize, log transform, filter and remove mitochondiral genes.",
                        "version": "0.4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00395%3A0.4",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00398:0.4",
                        "name": "STREAM.SeedEPGStructure",
                        "description": "Seeding the initial elastic principal graph prior to starting the trajectory learning process.",
                        "version": "0.4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00398%3A0.4",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00401:0.16",
                        "name": "STREAM.VisualizeTrajectories",
                        "description": "Generates flat tree subway and stream plots that intuitively represents trajectories as linear segments on a 2D plane",
                        "version": "0.16",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00401%3A0.16",
                        "categories": ["Single-Cell"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00101:4",
                        "name": "SubMap",
                        "description": "Maps subclasses between two data sets",
                        "version": "4",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00101%3A4",
                        "categories": ["Clustering"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00103:3",
                        "name": "SurvivalCurve",
                        "description": "Draws a survival curve based on a cls file",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00103%3A3",
                        "categories": ["Survival Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00104:1",
                        "name": "SurvivalDifference",
                        "description": "Tests for survival difference based on cls file",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00104%3A1",
                        "categories": ["Survival Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00025:4.1",
                        "name": "SVM",
                        "description": "Classify samples using SVM",
                        "version": "4.1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00025%3A4.1",
                        "categories": ["Prediction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00463:5",
                        "name": "Tangram",
                        "description": "A method for spatial alignment of single cell transcriptomic data. See <a href=https://github.com/genepattern/Tangram/blob/main/README.md>documentation<\/a> for details on use cases.",
                        "version": "5",
                        "documentation": "",
                        "categories": ["single-cell", "singlecell", "spatial analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00417:3",
                        "name": "TCGA.SampleSelection",
                        "description": "Retrieve TCGA data from Broad FireBrowse and perform sample selection on the basis of expression levels for specific genes of interest for analysis using GSEA tools.",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Agenepattern.org%3Amodule.analysis%3A00417%3A3",
                        "categories": ["preprocess & utilities", "gsea"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:8080.gpserver.ip-172-31-26-71.ip-172-31-26-71.ec2.internal:genepatternmodules:689:10",
                        "name": "TCGA.ssGSEA.ROC",
                        "description": "Pipeline for ssGSEA_ROC analysis of TCGA samples",
                        "version": "10",
                        "documentation": "",
                        "categories": ["pipeline", "gsea"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00369:6",
                        "name": "TCGAImporter",
                        "description": "This module imports data from TCGA by taking in a GDC manifest file, downloading the files listed on that manifest, renaming them to be human-friendly, and compiling them into a GCT file to be computer-friendly.",
                        "version": "6",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00369%3A6",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00443:1",
                        "name": "tfsites.FindTfSitesAlteredBySequenceVariation",
                        "description": "FindTfSitesAlteredBySequenceVariation reports the effects of all possible in silico single-nucleotide variants (SNVs) in a given sequence, for one transcription factor. Possible SNV effects include increasing (or optimizing) the affinity/score of a binding site, decreasing the affinity/score of a binding site, deleting a binding site, or creating a binding site.",
                        "version": "1",
                        "documentation": "",
                        "categories": ["Transcription factor analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00459:1",
                        "name": "tfsites.GenerateMotifDatabase",
                        "description": "GenerateMotifDatabase can be used to obtain position weight matrix (PWM) data to score binding sites in other TFSites modules. This tool extracts the PFM(s) for one or more transcription factor(s) of interest and then converts each PFM to its respective PWM.",
                        "version": "1",
                        "documentation": "",
                        "categories": ["Transcription factor analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00441:1",
                        "name": "tfsites.NormalizeTfDNAAffinityData",
                        "description": "NormalizeTfDnaAffinityData generates a relative affinity dataset which can then be used in other TFSites modules to score binding sites. This tool normalizes a raw affinity dataset relative to the sequence with the highest value that follows the core binding site definition. The resulting dataset will report the relative affinity value for every sequence in the original dataset, ranging from 0 to 1.",
                        "version": "1",
                        "documentation": "",
                        "categories": ["Transcription factor analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00442:1",
                        "name": "tfsites.VisualizeTFSitesOnSequences",
                        "description": "Annotates transcription factor binding sites across a DNA sequence. Multiple transcription factors can be analyzed. Each binding site is labeled with a unique binding site ID and its start and end position. If reference data is provided for a transcription factor, the affinity/score of this site will be labeled and the intensity of the binding site's color will be proporti",
                        "version": "1",
                        "documentation": "",
                        "categories": ["Transcription factor analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00249:1",
                        "name": "TmapToGct",
                        "description": "Converts a TMAP file to GCT format",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00249%3A1",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00026:3",
                        "name": "TransposeDataset",
                        "description": "Transpose a Dataset - .res .gct, .odf",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00026%3A3",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00341:2",
                        "name": "Trimmomatic",
                        "description": "Provides a variety of options for trimming Illumina FASTQ files of adapter sequences and low-quality reads.",
                        "version": "2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00341%3A2",
                        "categories": ["RNA-seq", "Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:genepattern.org:module.analysis:00422:5",
                        "name": "tximport.DESeq2",
                        "description": "Imports RNA-seq quantifications from frequently used pipelines using the tximport package and outputs a DESeq2 \"normalized counts\" file in .GCT format suitable for use with GSEA, and/or a TPM .GCT suitable for ssGSEA.<br>\nIf a Sample Info file is provided containing assignments of samples to phenotypes, gene level differential expression will be calculated using DESeq2 for the two phenotypes.",
                        "version": "5",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Agenepattern.org%3Amodule.analysis%3A00422%3A5",
                        "categories": ["differential expression"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00372:3",
                        "name": "txt2odf",
                        "description": "This module turns a TXT file into an ODF file which can be used with other GenePattern Modules.",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00372%3A3",
                        "categories": ["Data Format Conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00416:1.3",
                        "name": "txt2odf.20211130.colparams",
                        "description": "This module turns a TXT file into an ODF file which can be used with other GenePattern Modules.THIS MODULE IS FOR USE IN THE Dec 21, 2021 workshop and is in active development",
                        "version": "1.3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00416%3A1.3",
                        "categories": ["data format conversion"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00067:1",
                        "name": "UniquifyLabels",
                        "description": "Makes row and column labels unique",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00067%3A1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00216:0.1",
                        "name": "UploadFile",
                        "description": "A utility for users to upload and share a common input file for use by modules within a pipeline.",
                        "version": "0.1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00216%3A0.1",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00105:1",
                        "name": "VennDiagram",
                        "description": "Displays a Venn diagram",
                        "version": "1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.visualizer%3A00105%3A1",
                        "categories": ["Visualizer"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00355:1.2",
                        "name": "VoomNormalize",
                        "description": "Preprocess RNA-Seq count data in a GCT file so that it is suitable for use in GenePattern analyses. Formerly called \"PreprocessReadCounts\"",
                        "version": "1.2",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00355%3A1.2",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00027:3",
                        "name": "WeightedVoting",
                        "description": "Weighted Voting classification",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00027%3A3",
                        "categories": ["Prediction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00028:3",
                        "name": "WeightedVotingXValidation",
                        "description": "Weighted Voting classification with Leave-One-Out Cross-Validation",
                        "version": "3",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00028%3A3",
                        "categories": ["Prediction"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00093:3.1",
                        "name": "XChromosomeCorrect",
                        "description": "Corrects X Chromosome SNP's for male samples.",
                        "version": "3.1",
                        "documentation": "/gp/getTaskDoc.jsp?name=urn%3Alsid%3Abroad.mit.edu%3Acancer.software.genepattern.module.analysis%3A00093%3A3.1",
                        "categories": ["SNP Analysis"],
                        "suites": [],
                        "tags": []
                    }, {
                        "lsid": "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:99995:0.1",
                        "name": "ZipFiles",
                        "description": "Zips files",
                        "version": "0.1",
                        "documentation": "",
                        "categories": ["Preprocess & Utilities"],
                        "suites": [],
                        "tags": []
                    }],
                    "all_categories": [{
                        "name": "alternative splicing",
                        "description": "",
                        "tags": []
                    }, {"name": "batch correction", "description": "", "tags": []}, {
                        "name": "clustering",
                        "description": "",
                        "tags": []
                    }, {"name": "cnv analysis", "description": "", "tags": []}, {
                        "name": "data format conversion",
                        "description": "",
                        "tags": []
                    }, {"name": "differential expression", "description": "", "tags": []}, {
                        "name": "dimension reduction",
                        "description": "",
                        "tags": []
                    }, {"name": "ecdna", "description": "", "tags": []}, {
                        "name": "flow cytometry",
                        "description": "",
                        "tags": []
                    }, {"name": "gene list selection", "description": "", "tags": []}, {
                        "name": "gsea",
                        "description": "",
                        "tags": []
                    }, {"name": "image creators", "description": "", "tags": []}, {
                        "name": "javascript",
                        "description": "",
                        "tags": []
                    }, {"name": "metabolomics", "description": "", "tags": []}, {
                        "name": "methylation",
                        "description": "",
                        "tags": []
                    }, {
                        "name": "missing value imputation",
                        "description": "",
                        "tags": []
                    }, {
                        "name": "mutational significance analysis",
                        "description": "",
                        "tags": []
                    }, {"name": "pathway analysis", "description": "", "tags": []}, {
                        "name": "pipeline",
                        "description": "",
                        "tags": []
                    }, {"name": "prediction", "description": "", "tags": []}, {
                        "name": "preprocess & utilities",
                        "description": "",
                        "tags": []
                    }, {"name": "projection", "description": "", "tags": []}, {
                        "name": "proteomics",
                        "description": "",
                        "tags": []
                    }, {"name": "rna velocity", "description": "", "tags": []}, {
                        "name": "rna-seq",
                        "description": "",
                        "tags": []
                    }, {"name": "rnai", "description": "", "tags": []}, {
                        "name": "sage",
                        "description": "",
                        "tags": []
                    }, {"name": "sequence analysis", "description": "", "tags": []}, {
                        "name": "single cell gsea",
                        "description": "",
                        "tags": []
                    }, {"name": "single-cell", "description": "", "tags": []}, {
                        "name": "singlecell",
                        "description": "",
                        "tags": []
                    }, {"name": "singlecell gsea", "description": "", "tags": []}, {
                        "name": "snp analysis",
                        "description": "",
                        "tags": []
                    }, {"name": "spatial analysis", "description": "", "tags": []}, {
                        "name": "spatial transcriptomics",
                        "description": "",
                        "tags": []
                    }, {
                        "name": "spatial transcriptomics, javascript",
                        "description": "",
                        "tags": []
                    }, {"name": "statistical methods", "description": "", "tags": []}, {
                        "name": "survival analysis",
                        "description": "",
                        "tags": []
                    }, {"name": "transcription factor analysis", "description": "", "tags": []}, {
                        "name": "tutorial",
                        "description": "",
                        "tags": []
                    }, {"name": "uncategorized", "description": "", "tags": []}, {
                        "name": "variant annotation",
                        "description": "",
                        "tags": []
                    }, {"name": "viewer", "description": "", "tags": []}, {
                        "name": "visualizer",
                        "description": "",
                        "tags": []
                    }],
                    "all_suites": [],
                    "kindToModules": {
                        "fcs": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00176:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00198:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00179:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00268:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00182:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00181:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00180:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00175:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00192:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00185:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00210:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00186:2"],
                        "csv": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00198:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00049:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00172:9", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00405:5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00052:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00174:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00393:1.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00158:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00383:4", "urn:lsid:127.0.0.1:genepatternmodules:4:1.5.62", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00050:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00417:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00186:2", "urn:lsid:genepattern.org:module.analysis:00460:7"],
                        "zip": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00346:2", "urn:lsid:genepattern.org:module.analysis:00427:41", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00256:1.17", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00041:12.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00159:4.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00073:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00055:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00173:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00070:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00187:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00064:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00379:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00068:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00048:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15", "urn:lsid:7070.genepatt.genepatterntest.broadinstitute.org:genepatternmodules:268:6", "urn:lsid:genepattern.org:module.analysis:00416:4.11", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00086:3.12", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00095:0.3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00370:2"],
                        "txt": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00346:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00090:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00127:3.6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00409:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00085:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00335:0.21", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00158:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00041:12.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00340:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00125:6.15.30", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00412:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00083:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00094:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00173:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00187:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00354:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00092:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00107:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00380:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00320:1.3.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00166:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00411:1.3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00417:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:genepattern.org:module.analysis:00428:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00382:1.3", "urn:lsid:genepattern.org:module.analysis:00425:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00186:2", "urn:lsid:8080.gpserver.ip-172-31-26-71.ip-172-31-26-71.ec2.internal:genepatternmodules:941:1.3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00217:0.4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00086:3.12", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00302:1.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00106:1.3", "urn:lsid:genepattern.org:module.analysis:00460:7", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00300:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00103:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00104:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00369:6", "urn:lsid:genepattern.org:module.analysis:00459:1", "urn:lsid:genepattern.org:module.analysis:00422:5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00372:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00416:1.3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00105:1"],
                        "xz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00346:2"],
                        "gz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00346:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00042:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00055:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00379:4", "urn:lsid:genepattern.org:module.analysis:00416:4.11", "urn:lsid:genepattern.org:module.analysis:00460:7", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00395:0.4", "urn:lsid:genepattern.org:module.analysis:00422:5"],
                        "tar": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00346:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00055:2", "urn:lsid:genepattern.org:module.analysis:00416:4.11"],
                        "clm": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00346:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00041:12.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1"],
                        "bz2": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00346:2"],
                        "cel": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00346:2"],
                        "directory": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00346:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00335:0.21", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00326:0.30", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00342:0.11", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00336:0.17", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00275:0.3"],
                        "gmx": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00378:2", "urn:lsid:8080.gpbroad.broadinstitute.org:genepatternmodules:345:1.4.20", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00072:20.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00317:7.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00018:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:genepattern.org:module.analysis:00461:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00270:10.1.0", "urn:lsid:8080.gpserver.ip-172-31-26-71.ip-172-31-26-71.ec2.internal:genepatternmodules:689:10"],
                        "grp": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00378:2", "urn:lsid:8080.gpbroad.broadinstitute.org:genepatternmodules:345:1.4.20", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00072:20.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00317:7.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00018:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:genepattern.org:module.analysis:00461:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00270:10.1.0", "urn:lsid:8080.gpserver.ip-172-31-26-71.ip-172-31-26-71.ec2.internal:genepatternmodules:689:10", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00105:1"],
                        "gct": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00378:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00090:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00056:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00062:2.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00261:4.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00134:2.2.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00127:3.6", "urn:lsid:genepattern.org:module.analysis:00430:6.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00044:11", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00030:7.2", "urn:lsid:8080.gpbroad.broadinstitute.org:genepatternmodules:345:1.4.20", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00188:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00097:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00362:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00045:10", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00046:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00004:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00152:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00007:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00072:20.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00032:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00010:13.11", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00009:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00054:13", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00081:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00012:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00013:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00070:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00064:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00092:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00107:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00110:1", "urn:lsid:127.0.0.1:genepatternmodules:4:1.5.62", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00166:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00016:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00017:4.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00018:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00020:5.1", "urn:lsid:8080.gpserver.ip-172-31-26-71.ip-172-31-26-71.ec2.internal:genepatternmodules:69:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00417:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00315:1.10", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00033:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00021:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00022:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00217:0.4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00029:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00227:1.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00024:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00270:10.1.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00414:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00101:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00025:4.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00026:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00372:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00416:1.3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00067:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00355:1.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00027:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00028:3"],
                        "gmt": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00378:2", "urn:lsid:8080.gpbroad.broadinstitute.org:genepatternmodules:345:1.4.20", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00072:20.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00317:7.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00018:6", "urn:lsid:8080.gpserver.ip-172-31-26-71.ip-172-31-26-71.ec2.internal:genepatternmodules:69:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:genepattern.org:module.analysis:00437:4", "urn:lsid:genepattern.org:module.analysis:00461:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00270:10.1.0", "urn:lsid:8080.gpserver.ip-172-31-26-71.ip-172-31-26-71.ec2.internal:genepatternmodules:689:10"],
                        "bed": ["urn:lsid:genepattern.org:module.analysis:00427:41", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00251:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00379:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1"],
                        ".gz": ["urn:lsid:genepattern.org:module.analysis:00427:41", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00086:3.12", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00302:1.2"],
                        ".tar.gz": ["urn:lsid:genepattern.org:module.analysis:00427:41", "urn:lsid:genepattern.org:module.analysis:00429:11", "urn:lsid:genepattern.org:module.analysis:00426:1", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:genepattern.org:module.analysis:00420:0.9", "urn:lsid:genepattern.org:module.analysis:00416:4.11"],
                        "BAM": ["urn:lsid:genepattern.org:module.analysis:00427:41"],
                        ".fastq.gz": ["urn:lsid:genepattern.org:module.analysis:00427:41", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00383:4", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:genepattern.org:module.analysis:00419:0.5", "urn:lsid:genepattern.org:module.analysis:00420:0.9"],
                        "bam": ["urn:lsid:genepattern.org:module.analysis:00427:41", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00340:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00412:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00266:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00252:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00267:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00264:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00240:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00245:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15"],
                        ".zip": ["urn:lsid:genepattern.org:module.analysis:00429:11", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:genepattern.org:module.analysis:00463:5"],
                        "res": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00090:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00056:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00062:2.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00134:2.2.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00127:3.6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00044:11", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00030:7.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00188:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00097:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00045:10", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00003:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00046:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00004:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00007:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00072:20.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00032:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00010:13.11", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00009:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00081:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00012:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00013:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00015:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00070:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00064:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00107:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00110:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00017:4.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00018:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00020:5.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00315:1.10", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00033:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00021:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00022:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00029:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00024:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00025:4.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00026:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00067:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00027:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00028:3"],
                        "fastq.gz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00340:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00383:4", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:genepattern.org:module.analysis:00419:0.5", "urn:lsid:genepattern.org:module.analysis:00420:0.9", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00341:2"],
                        "DIRECTORY": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3"],
                        "fq.gz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00340:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:genepattern.org:module.analysis:00419:0.5", "urn:lsid:genepattern.org:module.analysis:00420:0.9", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00341:2"],
                        "fn": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00256:1.17"],
                        "fa": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00256:1.17", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00253:1.10", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:genepattern.org:module.analysis:00434:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00244:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00264:1", "urn:lsid:genepattern.org:module.analysis:00424:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00265:0.3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00370:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00371:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00341:2"],
                        "fq": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00256:1.17", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00340:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00370:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00341:2"],
                        "fa.gz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3", "urn:lsid:genepattern.org:module.analysis:00434:2", "urn:lsid:genepattern.org:module.analysis:00424:1", "urn:lsid:genepattern.org:module.analysis:00419:0.5"],
                        "fasta.gz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3", "urn:lsid:genepattern.org:module.analysis:00434:2", "urn:lsid:genepattern.org:module.analysis:00424:1"],
                        "fasta": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00256:1.17", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00253:1.10", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00368:1.9", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:genepattern.org:module.analysis:00434:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00244:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00264:1", "urn:lsid:genepattern.org:module.analysis:00424:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00265:0.3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00370:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00371:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00341:2"],
                        "fastq": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00254:2.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00256:1.17", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00340:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00383:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00241:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00370:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00341:2"],
                        "model": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00056:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00025:4.1"],
                        "cls": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00056:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00062:2.2", "urn:lsid:genepattern.org:module.analysis:00430:6.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00044:11", "urn:lsid:8080.gpbroad.broadinstitute.org:genepatternmodules:345:1.4.20", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00188:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00362:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00072:20.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00012:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00013:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00092:1", "urn:lsid:127.0.0.1:genepatternmodules:4:1.5.62", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00018:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00033:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00021:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00024:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00414:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00101:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00103:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00104:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00025:4.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00372:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00416:1.3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00355:1.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00027:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00028:3"],
                        "cn": ["urn:lsid:broadinstitute.org:cancer.genome.analysis:00057:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00125:6.15.30", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00070:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00064:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00021:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00022:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:broadinstitute.org:cancer.genome.analysis:00065:0.4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00084:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00300:1"],
                        "mtx": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00376:0.93", "urn:lsid:genepattern.org:module.analysis:00428:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00382:1.3"],
                        "rds": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00376:0.93", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00410:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00409:3", "urn:lsid:genepattern.org:module.analysis:00423:8", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00408:1.2", "urn:lsid:genepattern.org:module.analysis:00415:4.1", "urn:lsid:genepattern.org:module.analysis:00421:2", "urn:lsid:genepattern.org:module.analysis:00466:1", "urn:lsid:genepattern.org:module.analysis:00461:3", "urn:lsid:genepattern.org:module.analysis:00465:2", "urn:lsid:genepattern.org:module.analysis:00468:2", "urn:lsid:genepattern.org:module.analysis:00469:1", "urn:lsid:genepattern.org:module.analysis:00471:1"],
                        "chip": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00134:2.2.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00072:20.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00317:7.4.0", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:genepattern.org:module.analysis:00437:4"],
                        "odf": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00127:3.6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00045:10", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00010:13.11", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00081:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1"],
                        ".csv": ["urn:lsid:genepattern.org:module.analysis:00430:6.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00417:4"],
                        ".tsv": ["urn:lsid:genepattern.org:module.analysis:00430:6.1", "urn:lsid:genepattern.org:module.analysis:00426:1", "urn:lsid:genepattern.org:module.analysis:00443:1", "urn:lsid:genepattern.org:module.analysis:00441:1", "urn:lsid:genepattern.org:module.analysis:00442:1"],
                        "Dataset": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00030:7.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00097:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00046:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00004:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00007:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00032:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00010:13.11", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00009:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00012:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00013:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00070:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00064:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00017:4.2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00020:5.1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00033:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00021:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00022:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00029:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00026:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00067:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00027:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00028:3"],
                        "tar.gz": ["urn:lsid:genepattern.org:module.analysis:00426:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00055:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00379:4", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:genepattern.org:module.analysis:00420:0.9", "urn:lsid:genepattern.org:module.analysis:00416:4.11"],
                        "snp": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00085:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00094:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00070:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00064:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00021:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00022:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:broadinstitute.org:cancer.genome.analysis:00065:0.4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00096:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00084:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00093:3.1"],
                        "xls": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00085:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "db": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00335:0.21", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00326:0.30", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00342:0.11", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00336:0.17"],
                        "adj": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00097:2"],
                        "Comparative Marker Selection": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00045:10", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00046:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "Differential Expression": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00045:10"],
                        "cdf": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00041:12.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00086:3.12", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00302:1.2"],
                        "expr": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00248:2"],
                        "sam": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00340:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00412:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00266:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00267:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00264:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00234:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00240:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00245:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1"],
                        "Prediction Features": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00005:4"],
                        "fpkm_tracking": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00250:5"],
                        "Gene List": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00006:4"],
                        "soft": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00042:6"],
                        "mat": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00125:6.15.30"],
                        "seg": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00125:6.15.30", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00300:1"],
                        "glad": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00125:6.15.30", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1"],
                        "alf": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00125:6.15.30"],
                        "CNV": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00125:6.15.30"],
                        "rnk": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00317:7.4.0"],
                        ".rds": ["urn:lsid:genepattern.org:module.analysis:00435:6", "urn:lsid:genepattern.org:module.analysis:00437:4"],
                        "pcl": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00009:6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00151:1"],
                        "cdt": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00071:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00031:11.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00063:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1"],
                        "gtr": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00071:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00031:11.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00063:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1"],
                        "atr": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00071:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00031:11.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00063:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1"],
                        "fastq.bz2": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00341:2"],
                        "fq.bz2": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00341:2"],
                        "fasta.bz2": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3"],
                        "fa.bz2": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00367:3"],
                        "gtf": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00368:1.9", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00412:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00383:4", "urn:lsid:genepattern.org:module.analysis:00424:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00339:0.15", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00371:1", "urn:lsid:genepattern.org:module.analysis:00422:5"],
                        "gff3.gz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00412:4"],
                        "gff.gz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00412:4"],
                        "gff": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00412:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1"],
                        "gff3": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00412:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:genepattern.org:module.analysis:00422:5"],
                        "gtf.gz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00412:4", "urn:lsid:genepattern.org:module.analysis:00424:1", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:genepattern.org:module.analysis:00419:0.5"],
                        "bb": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "loh": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00070:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00064:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00021:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00022:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1"],
                        "psl": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "bai": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1"],
                        "fai": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15"],
                        "maf": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00380:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00320:1.3.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1"],
                        "Matrix": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00018:6"],
                        "wig": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1"],
                        "bw": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "xml": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "vcf": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00411:1.3"],
                        "cbs.txt": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "igv": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "sai": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1"],
                        "bedgraph": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "xcn": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00285:1", "urn:lsid:broadinstitute.org:cancer.genome.analysis:00065:0.4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00084:3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00300:1"],
                        "bigwig": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "bigbed": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "tdf": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "tbi": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        "idx": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00363:2.5"],
                        ".fa.gz": ["urn:lsid:genepattern.org:module.analysis:00434:2", "urn:lsid:genepattern.org:module.analysis:00424:1", "urn:lsid:genepattern.org:module.analysis:00419:0.5"],
                        "FASTA": ["urn:lsid:genepattern.org:module.analysis:00434:2", "urn:lsid:genepattern.org:module.analysis:00424:1"],
                        "tsv": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00383:4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00411:1.3", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:genepattern.org:module.analysis:00460:7", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00395:0.4", "urn:lsid:genepattern.org:module.analysis:00459:1", "urn:lsid:genepattern.org:module.analysis:00442:1", "urn:lsid:genepattern.org:module.analysis:00422:5"],
                        "FASTQ": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00383:4"],
                        "KNN Prediction Model": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00012:4"],
                        "Spectrum Peaks": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00051:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00053:2"],
                        "tgz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00055:2"],
                        "counts": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00354:1"],
                        "mindy": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00110:1"],
                        "msdata": ["urn:lsid:127.0.0.1:genepatternmodules:4:1.5.62"],
                        "mzXML": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00069:1"],
                        "tab": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00411:1.3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00371:1"],
                        "VCF": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00411:1.3"],
                        ".rda": ["urn:lsid:genepattern.org:module.analysis:00433:3"],
                        ".RDa": ["urn:lsid:genepattern.org:module.analysis:00433:3"],
                        "rdata": ["urn:lsid:genepattern.org:module.analysis:00433:3"],
                        "RData": ["urn:lsid:genepattern.org:module.analysis:00433:3"],
                        "fna": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00264:1"],
                        "dict": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00264:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15"],
                        "Prediction Results": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00019:4.11"],
                        ".gtf.gz": ["urn:lsid:genepattern.org:module.analysis:00424:1", "urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:genepattern.org:module.analysis:00419:0.5"],
                        "h5ad": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00417:4", "urn:lsid:genepattern.org:module.analysis:00428:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00382:1.3", "urn:lsid:genepattern.org:module.analysis:00425:2"],
                        "txt,gct": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00417:4"],
                        "csv, txt": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00417:4"],
                        ".pkl": ["urn:lsid:genepattern.org:module.analysis:00436:4"],
                        ".cls": ["urn:lsid:genepattern.org:module.analysis:00436:4"],
                        ".gct": ["urn:lsid:genepattern.org:module.analysis:00436:4"],
                        "read_group_tracking": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00339:0.15"],
                        "gct, res": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00184:1"],
                        "fq1": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00338:1"],
                        "list": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00263:2.15"],
                        ".fq.gz": ["urn:lsid:genepattern.org:module.analysis:00418:0.6", "urn:lsid:genepattern.org:module.analysis:00419:0.5", "urn:lsid:genepattern.org:module.analysis:00420:0.9"],
                        "loom": ["urn:lsid:genepattern.org:module.analysis:00428:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00382:1.3", "urn:lsid:genepattern.org:module.analysis:00416:4.11"],
                        "h5": ["urn:lsid:genepattern.org:module.analysis:00428:2", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00382:1.3", "urn:lsid:genepattern.org:module.analysis:00422:5"],
                        "bsnp": ["urn:lsid:broadinstitute.org:cancer.genome.analysis:00065:0.4"],
                        ".h5ad": ["urn:lsid:genepattern.org:module.analysis:00437:4", "urn:lsid:genepattern.org:module.analysis:00463:5"],
                        ".h5seurat": ["urn:lsid:genepattern.org:module.analysis:00437:4"],
                        "SOM Cluster": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.visualizer:00023:5.1"],
                        "json": ["urn:lsid:genepattern.org:module.analysis:00468:2", "urn:lsid:genepattern.org:module.analysis:00471:1"],
                        "seg.txt": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00300:1"],
                        "pkl": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00402:0.12", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00403:0.12", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00404:0.3", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00397:0.7", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00399:0.4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00400:0.6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00396:0.6", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00406:0.15", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00395:0.4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00398:0.4", "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00401:0.16"],
                        "tsv.gz": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00395:0.4"],
                        ".gmt": ["urn:lsid:genepattern.org:module.analysis:00463:5"],
                        ".txt": ["urn:lsid:genepattern.org:module.analysis:00443:1"],
                        "JASPAR": ["urn:lsid:genepattern.org:module.analysis:00459:1"],
                        "jaspar": ["urn:lsid:genepattern.org:module.analysis:00459:1"],
                        "tmap": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00249:1"],
                        "sf": ["urn:lsid:genepattern.org:module.analysis:00422:5"],
                        "results": ["urn:lsid:genepattern.org:module.analysis:00422:5"],
                        "Weighted Voting Prediction Model": ["urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00027:3"]
                    }
                };
            }

            build_dropdown(payload);
            build_table(payload);

            const param_filter = get_filter();
            if (param_filter) filter_modules(param_filter);
        }

        $(document).ready(() => {
            load_modules();
        });
    </script>
{% endblock %}