<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>BWA.aln (v2) BETA</title>
    <meta name="slug" content="modules/docs/BWA.aln/2" />
    <meta name="template" content="module_doc" />
</head>
<body>
<div class="bs-callout bs-callout-danger">
<h4>This module is currently in beta release.  The module and/or documentation may be incomplete.</h4>
</div>
<p style="font-size: 1.2em; margin-top: 20px;">A fast and accurate short-read alignment tool that allows for mismatches and gaps. Alignments are output in a SAM format file, which provides Phred-scale quality scores for each alignment.</p>
<div class="row">
<div class="col-sm-4">
<p><strong>Author: </strong>Heng Li, Broad Institute</p>
</div>
<div class="col-sm-4">
<p><strong>Contact: </strong><p>
	gp-help@broadinstitute.org</p>
</p>
</div>
<div class="col-sm-4">
<p><strong>Algorithm Version: </strong>BWA 0.7.4</p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Summary</h2>
<p>
	Burrows-Wheeler Aligner (BWA.aln) is a fast, light-weight tool that aligns relatively short nucleotide sequences against a long reference sequence such as the human genome. It works for query sequences shorter than 200bp, and does gapped alignment. BWA.aln is usually faster and more accurate on queries with low error rates.</p>
<p>
	This document is adapted from the BWA documentation for release 0.7.4.  For more information about BWA.aln, see the <a href="http://bio-bwa.sourceforge.net/">BWA project site</a>. BWA.aln was developed at the Wellcome Trust Sanger Institute and the Broad Institute.</p>
<p>
<strong style="font-size: 13px; color: rgb(34, 34, 34);">Note: </strong><span style="color: rgb(34, 34, 34);">Index files created with BWA version 0.5.x or earlier are </span><strong style="font-size: 13px; color: rgb(34, 34, 34);">not</strong><span style="color: rgb(34, 34, 34);"> compatible with the aligners of version 0.6.x and newer.  Likewise, the BWA 0.6.x and newer index files are not compatible with the 0.5.x aligners.  The BWA 0.7.x aligners are able to use index files created with 0.6.x, however.</span></p>
<h2>
	Speed</h2>
<p>
	Speed of alignment is largely determined by the error rate of the query sequences, faster with near-perfect hits and slower for higher error rates. Pairing is slower for shorter reads, mostly because shorter reads have more spurious hits.</p>
<p>
	In experimental runs, BWA was able to map 2 million 32bp reads to:</p>
<ul>
<li>
		a bacterial genome in several minutes</li>
<li>
		the human X chromosome in 8-15 minutes</li>
<li>
		the human genome in 15-25 minutes</li>
</ul>
<h2>
	References</h2>
<p>
	BWA manual page: <a href="http://bio-bwa.sourceforge.net/bwa.shtml">http://bio-bwa.sourceforge.net/bwa.shtml</a>.</p>
<p>
	Li H, Durbin R. Fast and accurate short read alignment with Burrows-Wheeler Transform. <em>Bioinformatics</em>. 2009;25:1754-1760. [PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19451168">19451168</a>] (<a href="http://www.ncbi.nlm.nih.gov/pubmed/19451168">http://www.ncbi.nlm.nih.gov/pubmed/19451168</a>)</p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Parameters</h2>
<table id="module_params_table">
<thead>
<tr>
<th>
				Name</th>
<th>
				Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>
				BWA index <span style="color: rgb(255, 0, 0);">*</span></td>
<td>
				A BWA index. You can select from a list of hosted indexes or provide a custom index in the form of a ZIP bundle (as generated by the BWA.indexer module).</td>
</tr>
<tr>
<td>
				reads pair 1 <span style="color:red;">*</span></td>
<td>
				Single-end or first paired-end reads file in FASTA, FASTQ, or BAM format.  For paired-end data, this should be the forward ("*_1" or "left") input file.  Note: the FASTA or FASTQ can be gzipped.</td>
</tr>
<tr>
<td>
				reads pair 2</td>
<td>
				The reverse ("*_2" or "right") input file for paired-end reads in FASTA, FASTQ, or BAM format. Note: the FASTA or FASTQ can be gzipped.</td>
</tr>
<tr>
<td>
				bam mapping</td>
<td>
				Specifies how to map BAM input. This is only required if input file is in BAM format.</td>
</tr>
<tr>
<td>
				max edit distance</td>
<td>
				The max edit distance. This specifies a threshold of the maximum number of deletions, insertions, and substitutions needed to transform the reference sequence into the read sequence.</td>
</tr>
<tr>
<td>
				max num gap</td>
<td>
				Maximum number of gap opens. This specifies a threshold of the maximum number of gaps that can be initiated to match a given read to the reference.</td>
</tr>
<tr>
<td>
				max gap extension</td>
<td>
				Maximum number of gap extensions. This specifies a threshold of the maximum number of bases by which gaps in a read can be extended.</td>
</tr>
<tr>
<td>
				max deletion length</td>
<td>
				Disallow a long deletion within this many bp of the 3' end.</td>
</tr>
<tr>
<td>
				max indel length</td>
<td>
				Disallow an indel within this many bp of the ends</td>
</tr>
<tr>
<td>
				seed length</td>
<td>
				The set of bases determined by this option in the high-quality (left) end of the read is the seed.</td>
</tr>
<tr>
<td>
				max seed edit distance</td>
<td>
				Maximum edit distance in the seed; that is, the maximum number of changes required to transform the reference sequence of the seed into the read sequence of the seed.</td>
</tr>
<tr>
<td>
				mismatch penalty</td>
<td>
				Mismatch penalty</td>
</tr>
<tr>
<td>
				gap open penalty</td>
<td>
				Gap open penalty.  The gap open penalty is the score taken away for the initiation of the gap in sequence. To make the match more significant you can try to make the gap penalty larger.</td>
</tr>
<tr>
<td>
				gap extension penalty</td>
<td>
				Gap extension penalty.  The gap extension penalty is added to the standard gap penalty for each base or residue in the gap.  To reduce long gaps, increase the extension gap penalty. A few long gaps are expected, rather than many short gaps, so the gap extension penalty should be lower than the gap penalty. (The exception to this rule is where one or both sequences are single reads with possible sequencing errors, in which case many single base gaps are expected. To cope with this, try setting the gap open penalty very low and using the gap extension penalty to control gap scoring.)</td>
</tr>
<tr>
<td>
				max best hits</td>
<td>
				Proceed with suboptimal alignments if there are no more than this many equally best hits. This option only affects paired-end mapping.  Increasing this threshold helps to improve the pairing accuracy at the cost of speed, especially for short reads (~32 bp).</td>
</tr>
<tr>
<td>
				iterative search <span style="color:red;">*</span></td>
<td>
				Whether to disable iterative search.  Enabling this will slow the alignment process.</td>
</tr>
<tr>
<td>
				trim reads</td>
<td>
				Specifies a quality threshold for read trimming.  The trimming algorithm in BWA scans from the right of the read, accumulating a penalty sum (or "area") for each position that is lower quality than this threshold and reducing this area for each position above that threshold.  The read is trimmed to the position where the penalty area is greatest.</td>
</tr>
<tr>
<td>
				Illumina 1 3 format <span style="color:red;">*</span></td>
<td>
				The input is in the Illumina 1.3+ read format</td>
</tr>
<tr>
<td>
				barcode length</td>
<td>
				Length of barcode starting from the 5' end</td>
</tr>
<tr>
<td>
				max insert size</td>
<td>
				Specifies the maximum insert size for a read pair to be considered to be mapped properly</td>
</tr>
<tr>
<td>
				max occurrences</td>
<td>
				Specifies the maximum occurrences of a read for pairing</td>
</tr>
<tr>
<td>
				max alignments</td>
<td>
				Maximum number of alignments to output in the XA tag for reads paired properly</td>
</tr>
<tr>
<td>
				max dc alignments</td>
<td>
				Maximum number of alignments to output in the XA tag for disconcordant read pairs (excluding singletons)</td>
</tr>
<tr>
<td>
				output prefix <span style="color:red;">*</span></td>
<td>
				Prefix to use for output file name</td>
</tr>
</tbody>
</table>
<p>
<span style="color: red;">*</span> - required</p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Input Files</h2>
<ol>
<li>
		BWA index<br/>
		A set of BWA index files bundled as a ZIP archive, as produced by the BWA.indexer module.  The GenePattern FTP site also hosts a number of index bundels, available in a dropdown selection (requires GenePattern 3.7.0+).<br/>
<strong>Note:</strong> these index files must have been produced by BWA version 0.6.x or 0.7.x.</li>
<li>
		reads pair 1<br/>
		Single-end or first paired-end reads file in FASTA, FASTQ, or BAM format.  For paired-end data, this should be the forward ("*_1" or "left") input file.  Note: the FASTA or FASTQ can be gzipped.</li>
<li>
		reads pair 2<br/>
		The reverse ("*_2" or "right") input file for paired-end reads in FASTA, FASTQ, or BAM format. Note: the FASTA or FASTQ can be gzipped.</li>
</ol>
<h2>
	Output Files</h2>
<ol>
<li>
		SAM file<br/>
		The aligned sequences are output in SAM format. For more details on this alignment file, see the SAM format specification at <a href="http://samtools.sourceforge.net/SAM-1.3.pdf">http://samtools.sourceforge.net/SAM-1.3.pdf</a>.</li>
</ol>
</div>
</div>
<h2>Platform Dependencies</h2>
<div class="row">
<div class="col-md-3">
<p><strong>Task Type:</strong><br/>
					RNA-seq</p>
</div>
<div class="col-md-3">
<p><strong>CPU Type:</strong><br/>
					any</p>
</div>
<div class="col-md-3">
<p><strong>Operating System:</strong><br/>
					any</p>
</div>
<div class="col-md-3">
<p><strong>Language:</strong><br/>
					C;Perl</p>
</div>
</div>
<h2>Version Comments</h2>
<table class="table table-striped">
<thead>
<tr>
<th>Version</th>
<th>Release Date</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>2</td>
<td>2014-07-17</td>
<td>Beta: Updated to BWA 0.7.4, changed to use dynamic FTP-hosted index files, and switched to HTML-based doc</td>
</tr>
<tr>
<td>1</td>
<td>2011-05-02</td>
<td></td>
</tr>
</tbody>
</table>
</body>
</html>