<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>BEDTools.intersect (v1) BETA</title>
    <meta name="slug" content="modules/docs/BEDTools.intersect/1" />
    <meta name="template" content="module_doc" />
</head>
<body>
<div class="bs-callout bs-callout-danger">
<h4>This module is currently in beta release.  The module and/or documentation may be incomplete.</h4>
</div>
<p style="font-size: 1.2em; margin-top: 20px;">Screen for overlaps between two sets of genomic features.</p>
<div class="row">
<div class="col-sm-4">
<p><strong>Author: </strong>Chet Birger, Broad Institute</p>
</div>
<div class="col-sm-4">
<p><strong>Contact: </strong>gp-help@broadinstitute.org</p>
</div>
<div class="col-sm-4">
<p><strong>Algorithm Version: </strong></p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Introduction</h2>
<p>
	BEDtools.intersect allows one to screen for overalaps between two sets of genomic features.  It allows one to have fine control as to how the intersections are reported.  The input files describing the two sets of genomic features may be BED/GFF/VCF or BAM files.</p>
<h2>
	References</h2>
<p>
<a href="http://bedtools.readthedocs.org/en/latest/content/tools/intersect.html" style="color: rgb(102, 153, 255);">BEDTools intersect online documentation</a></p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Parameters</h2>
<table id="module_params_table">
<thead>
<tr>
<th>
				Name</th>
<th>
				Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>
				input file A </td>
<td>
				BED/GFF/VCF file A containing first set of genomic features. (-a)</td>
</tr>
<tr>
<td>
				bam input file A </td>
<td>
				BAM file A. Each BAM alignment in A is compared to B in search of overlaps. (-abam)</td>
</tr>
<tr>
<td>
				input file B </td>
<td>
				BED/GFF/VCF file B containing second set of genomic features. (-b)</td>
</tr>
<tr>
<td>
				unique </td>
<td>
				Report the presence of any (at least one) overlapping feature. (-u)</td>
</tr>
<tr>
<td>
				no overlap </td>
<td>
				Only report those entries in A that have no overlap in B. (-v)</td>
</tr>
<tr>
<td>
				count </td>
<td>
				Report the number of genes that each alignment overlaps. (-c)</td>
</tr>
<tr>
<td>
				original A </td>
<td>
				Report the orignal A feature. (-wa)</td>
</tr>
<tr>
<td>
				original B </td>
<td>
				Report the original B feature. (-wb)</td>
</tr>
<tr>
<td>
				min overlap </td>
<td>
				Minimum overlap required as a fraction of A. Default is 1E-9 (i.e. 1bp). (-f)</td>
</tr>
<tr>
<td>
				reciprocal </td>
<td>
				Require that the fraction of overlap be reciprocal for A and B. In other words, if -f is 0.90 and -r is used, this requires that B overlap at least 90% of A and that A also overlaps at least 90% of B. (-r)</td>
</tr>
<tr>
<td>
				bed output </td>
<td>
				When comparing alignments in BAM format (-abam) to features in BED format (-b), bedtools intersect will optionally write the output in BED format if set to true. (-bed)</td>
</tr>
<tr>
<td>
				header </td>
<td>
				Print the header from the A file prior to results. (-header)</td>
</tr>
<tr>
<td>
				uncompressed BAM </td>
<td>
				Write uncompressed BAM output. The default is write compressed BAM output. (-ubam)</td>
</tr>
<tr>
<td>
				left outer join </td>
<td>
				Perform a left outer join. (-loj)</td>
</tr>
<tr>
<td>
				write overlap size </td>
<td>
				Write size of overlap (in bases); choose between no, intersecting (-wo), all (-wao)</td>
</tr>
<tr>
<td>
				strandedness </td>
<td>
				Select strandedness constraints. (-s, -S)</td>
</tr>
<tr>
<td>
				split </td>
<td>
				Treat split BAM (i.e., having an N CIGAR operation) or BED12 entries as distinct BED intervals. (-split)</td>
</tr>
<tr>
<td>
				sorted </td>
<td>
				For very large B files, invoke a sweeping algorithm that requires position-sorted (e.g., sort -k1,1 -k2,2n for BED files) input. When using -sorted, memory usage remains low even for very large files. (-sorted)</td>
</tr>
<tr>
<td>
				output filename <span style="color:red;">*</span></td>
<td>
				Output filename</td>
</tr>
</tbody>
</table>
<p>
<span style="color: red;">*</span> - required</p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Input Files</h2>
<ol>
<li style="padding-bottom: 8px;">
		input file A<br/>
		The first of two BED-, GFF- or VCF-formatted files containing annotation track data.  BEDTools.intersect computes the overlap of features in input file A with the features in input file B.</li>
<li style="padding-bottom: 8px;">
		input file B<br/>
		The second of two BED-, GFF- or VCF-formated files containing annotation track data.  BEDTools.interesect computes the overlap of features in input file A with the features in input file B.</li>
</ol>
<h2>
	Output Files</h2>
<ol>
<li>
		output filename<br/>
		See <a href="http://bedtools.readthedocs.org/en/latest/content/tools/intersect.html" style="color: rgb(102, 153, 255);">BEDTools intersect online documentation</a> for a description of various output reporting formats.</li>
</ol>
<h2>
	 </h2>
</div>
</div>
<h2>Platform Dependencies</h2>
<div class="row">
<div class="col-md-3">
<p><strong>Task Type:</strong><br/>
					ChIP-Seq</p>
</div>
<div class="col-md-3">
<p><strong>CPU Type:</strong><br/>
					any</p>
</div>
<div class="col-md-3">
<p><strong>Operating System:</strong><br/>
					any</p>
</div>
<div class="col-md-3">
<p><strong>Language:</strong><br/>
					C++</p>
</div>
</div>
<h2>Version Comments</h2>
<table class="table table-striped">
<thead>
<tr>
<th>Version</th>
<th>Release Date</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>2013-12-23</td>
<td>Initial Beta Version</td>
</tr>
</tbody>
</table>
</body>
</html>