<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>BEDTools.merge (v1) BETA</title>
    <meta name="slug" content="modules/docs/BEDTools.merge/1" />
    <meta name="template" content="module_doc" />
</head>
<body>
<div class="bs-callout bs-callout-danger">
<h4>This module is currently in beta release.  The module and/or documentation may be incomplete.</h4>
</div>
<p style="font-size: 1.2em; margin-top: 20px;">Combine overlapping or book-ended features in an interval file into a single feature which spans all of the combined features.</p>
<div class="row">
<div class="col-sm-4">
<p><strong>Author: </strong>Chet Birger, Broad Institute</p>
</div>
<div class="col-sm-4">
<p><strong>Contact: </strong>gp-help@broadinstitute.org</p>
</div>
<div class="col-sm-4">
<p><strong>Algorithm Version: </strong></p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Introduction</h2>
<p>
<tt class="docutils literal"><span class="pre">bedtools.</span><span class="pre">merge</span></tt> combines overlapping or “book-ended” features in a BED/GFF/VCF feature file into a single feature which spans all of the combined features.  <tt class="docutils literal" style="font-size: 12px; line-height: 18px;"><span class="pre">bedtools.</span><span class="pre">merge</span></tt><span style="font-size: 12px; line-height: 18px;"> requires that you presort your data by chromosome and then by start position.</span></p>
<h2>
	References</h2>
<p>
<a href="http://bedtools.readthedocs.org/en/latest/content/tools/merge.html" style="color: rgb(102, 153, 255);">BEDTools merge online documentation</a></p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Parameters</h2>
<table id="module_params_table">
<thead>
<tr>
<th>
				Name</th>
<th>
				Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>
				input file <span style="color:red;">*</span></td>
<td>
				Input BED/GFF/VCF file (-i)</td>
</tr>
<tr>
<td>
				force strandedness </td>
<td>
				Force strandedness; that is, only merge features that are on the same strand. (-s)</td>
</tr>
<tr>
<td>
				report number entries merged </td>
<td>
				Report the number of entries that were merged. 1 is reported if no merging occurred. (-n)</td>
</tr>
<tr>
<td>
				max distance for merge </td>
<td>
				Maximum distance between features (num base pairs) allowed for features to be merged. Default is 0. (-d)</td>
</tr>
<tr>
<td>
				report scores </td>
<td>
				Report scores of the merged features (-scores)</td>
</tr>
<tr>
<td>
				names </td>
<td>
				Report the names of the merged features separated by semicolons. (-nms)</td>
</tr>
<tr>
<td>
				output filename <span style="color:red;">*</span></td>
<td>
				Output filename</td>
</tr>
</tbody>
</table>
<p>
<span style="color: red;">*</span> - required</p>
</div>
</div>
<div class="row">
<div class="col-md-12">
<h2>
	Input Files</h2>
<ol>
<li>
		input file<br/>
		Input BED/GFF/VCF file </li>
</ol>
<h2>
	Output Files</h2>
<ol>
<li>
		output filename<br/>
		Contents of output file are determined by the option selection; see <a href="http://bedtools.readthedocs.org/en/latest/content/tools/merge.html" style="color: rgb(102, 153, 255);">BEDTools merge online documentation</a> for descriptions of these output formats.</li>
</ol>
</div>
</div>
<h2>Platform Dependencies</h2>
<div class="row">
<div class="col-md-3">
<p><strong>Task Type:</strong><br/>
					ChIP-Seq</p>
</div>
<div class="col-md-3">
<p><strong>CPU Type:</strong><br/>
					any</p>
</div>
<div class="col-md-3">
<p><strong>Operating System:</strong><br/>
					any</p>
</div>
<div class="col-md-3">
<p><strong>Language:</strong><br/>
					C++</p>
</div>
</div>
<h2>Version Comments</h2>
<table class="table table-striped">
<thead>
<tr>
<th>Version</th>
<th>Release Date</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>2013-12-23</td>
<td>Initial Beta version</td>
</tr>
</tbody>
</table>
</body>
</html>