<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Press Releases</title>
</head>
<body>
<p>Below are archived copies of previous GenePattern press releases. These press releases predate the current GenePattern <a href="http://www.broadinstitute.org/blog">blog</a>.</p>

<div class="well well-lg">
<h3>RNA-seq: Comparing strand-specific RNA sequencing techniques</h3>

<h4>dUTP protocol emerges as the leader</h4>

<p><strong>By Alice McCarthy, Broad Communications<br />
Published August 16, 2010</strong></p>

<div class="col-sm-3 pull-right"><img alt="RNA seq" class="img-thumbnail" src="/uploaded/content_nature_methods_proposal_1.png" style="width: 300px; height: 200px;" />
<p class="text-muted"><small>Researchers compared seven RNA-seq methods. The dUTP technique emerged as the leader.<br />
<i>Image courtesy of Sigrid Hart</i></small></p>
</div>

<p>We value having options in biomedical research. But sometimes having many choices, and without sufficient comparative information about their benefits and limitations, can unnecessarily complicate research progress. Such has been the situation over the past two years regarding the technology of complementary DNA (cDNA) &quot;second-generation&quot; sequencing, or &quot;RNA-seq&quot; as it has come to be known.</p>

<p>Complementary DNA is created in the laboratory starting from an RNA template using an enzyme called reverse transcriptase. The cDNA sequence is complementary to the sequence of its RNA template, hence its name. Over the past few years, researchers have developed a variety of techniques to decipher this cDNA and learn more about an organism from a cell&#39;s RNA content.<a href="http://www.nature.com/nmeth/journal/vaop/ncurrent/abs/nmeth.1491.html"> In a paper published August 15 in <i>Nature Methods</i></a>, researchers at the Broad Institute of Harvard and MIT directly compared seven of these methods, known as RNA-seq techniques. Using a set of criteria, a technique known as dUTP second-strand marking emerged as the leading protocol and has been adopted at the Broad for RNA-seq applications. Full details of this protocol are described in this paper <a href="http://nar.oxfordjournals.org/cgi/content/full/37/18/e123">and in the original 2009 dUTP paper by Parkhomchuk et al. in Nucleic Acids Research last year</a>. The authors also provide a menu of comparative analysis criteria that can be applied for assessment of future RNA-seq protocols.</p>

<p>Researchers perform RNA-seq for three general reasons. &quot;First, you want to know how many and which RNA transcripts are in a cell or in a sample,&quot; explains Joshua Levin, of the Genome Sequencing and Analysis Program (GSAP) at the Broad and co-first author of the paper. &quot;With RNA-seq, you can actually count the relative number of transcripts made in each cell, which tells you something about its function.&quot;</p>

<p>Second, RNA-seq provides specific information for genome annotation. &quot;It lets you identify the elements (DNA sequences) of the genome that are copied into RNA and assign them biological function,&quot; says Levin. In past years, genome annotation was done with expressed sequence tags (ESTs) but this relies on the older Sanger-based sequencing technology, which has rapidly been displaced by newer, second-generation techniques such as those provided by Illumina and others. Second-generation technologies have vastly increased access to the information-dense transcriptome.</p>

<p>Last, RNA-seq allows researchers to characterize RNA splicing - modifications of RNA after transcription, in which introns (nucleotide bases that are not expressed into proteins) are removed and exons (bases that are expressed) are joined. &quot;There are programs that can predict that but you really want to have an actual experiment that tells you what sequences are present in spliced RNAs and those that are not,&quot; says Levin. Differences in RNA splicing can lead to alterations in the proteins translated from those RNAs thereby imparting functional consequences for cells and organisms.</p>

<p>Levin&#39;s group worked with various RNA-seq methods as they became available, including two developed internally at the Broad. &quot;Though we&#39;ve been working with these techniques for several years, we realized that no one had compared them to determine which would be the best to recommend,&quot; says Levin. &quot;A lot of people are just getting started on RNA-seq so they don&#39;t know which method they should use.&quot; Researchers at the Broad often field technique-related questions from other investigators. This analysis was done largely to help the larger community sift through the options regarding RNA-seq.</p>

<p>In the cell, each single-stranded RNA is synthesized from one of the two strands of DNA. When RNA is copied back into cDNA for RNA-seq in the lab, the information about which of the two strands of DNA was copied into RNA can be lost unless special methods are used. The crux of this paper is to test which of seven different &quot;strand-specific&quot; methods is best to preserve this strand information. Strand-specific RNA-seq improves on standard RNA-seq in three ways: accurately identifying antisense transcripts, determining the transcribed strand of non-coding RNAs (e.g. lincRNAs), and demarcating the boundaries of closely situated or overlapping genes.</p>

<p>&quot;Nonstrand-specific RNA sequencing has been the standard method,&quot; explains Levin. &quot;But now strand-specific approaches provide additional valuable information and do not involve that much more work or cost.&quot;</p>

<p>Along with strand specificity, the team examined other criteria using their new computational pipeline. And they assessed practical measures like ease of use in the laboratory and in computational analysis. &quot;Looking at all these factors, dUTP turned out to be the one we liked the most and it is our default RNA-seq method at the Broad right now,&quot; says Levin. But he notes that technical challenges need to be addressed to make the process high-throughput. &quot;This technique works for making 12 libraries, for example,&quot; he says. &quot;But if you want to automate it, for 100 or more libraries at a time, the method needs to be modified.&quot; He comments that researchers at the Broad are addressing this point now to be ready for large sequencing requests as they become more frequent.</p>

<p>The team&#39;s analysis is freely available on the <a href="https://cloud.genepattern.org">GenePattern server</a>. &quot;This was done so that other researchers can evaluate their own protocols using the same criteria explained in the paper,&quot; explains Moran Yassour of the Broad Institute and the Hebrew University, and a co-first author of the paper.</p>

<p>Paper(s) cited:</p>

<p>Joshua Z Levin, Moran Yassour, et al. Comprehensive comparative analysis of strand-specific RNA sequencing methods. <i>Nature Methods</i>. 15 August 2010.<a href="http://www.nature.com/nmeth/journal/vaop/ncurrent/abs/nmeth.1491.html"> doi:10.1038/NMETH.1491</a></p>
<!-- Article #2 --></div>

<div class="well well-lg">
<h3>Broad Institute researchers expand the capabilities of GenePattern software</h3>

<p><strong>By Nicole Davis, Communications<br />
Published April 27, 2006</strong></p>

<p>Scientists at the Broad Institute of MIT and Harvard recently released GenePattern 2.0, an enhanced version of the integrative software tool for analyzing gene expression data. As described in the May issue of <i>Nature Genetics,</i> GenePattern provides a variety of analytic and visualization functions that enable researchers to perform custom data analyses and to record them for later playback. The updated software includes several new components that permit the analysis of proteomic data and improve its ability to capture and recall the individual steps in the analytic process. GenePattern is freely available to the research community and provides multiple user interfaces, which accommodate the needs of researchers with programming experience as well as those without it.</p>

<p>&quot;The strengths that GenePattern brings to gene expression analysis can now be similarly realized for proteomic data,&quot; said Michael Reich, lead author of the <i>Nature Genetics</i> letter, manager of cancer informatics development at the Broad Institute, and the group leader for GenePattern. &quot;We have also added features to improve its ability to capture and reproduce analyses, which is vital to researchers both individually and as a community.&quot;</p>

<p>The new additions to GenePattern enable researchers to process proteomic data and include modules for data processing, analysis, and visualization. Other enhancements supplement its reproducibility features, which include automatically recording the individual steps in an analytic process so they can be repeated or shared with other researchers, and storing both the data and all versions of the methodologies used to manipulate them. These aspects of GenePattern are key facilitators for replicating <i>in silico</i> research findings by independent researchers.</p>

<p>&quot;GenePattern meets the ever-changing needs of researchers in the age of genomic science,&quot; said Jill Mesirov, senior author of the <i>Nature Genetics</i> letter, chief informatics officer, and director of Computational Biology and Bioinformatics at the Broad Institute. &quot;These new capabilities are a vivid illustration of the tool&#39;s flexibility and adaptability, as well as our commitment to the goal of reproducible research.&quot;</p>

<p>In the future, scientists plan to incorporate additional improvements to the GenePattern software. These include capabilities for analyzing single nucleotide polymorphism (SNP) data, such as copy number estimation, loss of heterozygosity determination, and the identification of chromosomal amplifications and deletions. This information forms the core components for discovering the genomic alterations that contribute to cancer and other human diseases.</p>

<p>Currently, there are over 2,300 registered GenePattern users worldwide, including more than 500 institutions and 30 pharma-biotech companies. In 2005, GenePattern received the Editor&#39;s Choice award in the Bio-IT World Best Practices competition.</p>

<p>The updated software and a comprehensive list of new features and fixes can be found on the <a href="/" target="_blank">GenePattern</a> website.</p>

<p>GenePattern is supported by funding from the National Institutes of Health.</p>

<p>Reich M<sup>1</sup>, Liefeld T<sup>1</sup>, Gould J<sup>1</sup>, Tamayo P<sup>1</sup>, Mesirov JP<sup>1</sup>. <a href="http://www.nature.com/ng/journal/v38/n5/full/ng0506-500.html" target="_blank">GenePattern 2.0</a> <i>Nature Genetics;</i> doi:10.1038/ng1785</p>

<p><sup>1</sup>Broad Institute of MIT and Harvard, Cambridge, MA 02142</p>
</div>
<!-- Article #3 -->

<div class="well well-lg">
<h3>GenePattern receives Bio-IT World Best Practices award</h3>

<p><strong>By Broad Institute Communications<br />
Published July 6, 2005</strong></p>

<div class="col-sm-3 pull-right"><img alt="The GenePattern team: (l-r) Charlotte Henson, Josh Gould, Jill Mesirov, Michael Reich, Ted Liefeld, and Pablo Tamayo. Missing from the photo: Gad Getz, Jim Lerner, Stefano Monti, Ken Ross, and Aravind Subramanian" class="img-thumbnail" height="225" src="http://www.broadinstitute.org/files/news/stories/full/GPaward-0706.jpg" width="300" />
<p class="text-muted"><small>The GenePattern team<br />
<i>Left to right:</i> Charlotte Henson, Josh Gould, Jill Mesirov, Michael Reich, Ted Liefeld, and Pablo Tamayo.<br />
<i>Not pictured</i>: Gad Getz, Jim Lerner, Stefano Monti, Ken Ross, and Aravind Subramanian</small></p>
</div>

<p>GenePattern, a gene expression analysis software package developed by researchers from the Broad Institute, was chosen for an Editor&#39;s Choice award at the 2005 <i>Bio-IT World</i> Best Practices celebration on June 28. The freely available application was selected for this award from 33 different submissions by a panel of computational biology experts.</p>

<p>&quot;We are thrilled by this honor. Improving research practices is one of GenePattern&#39;s basic goals,&quot; said Michael Reich, manager of cancer informatics development and group leader for GenePattern. &quot;As this tool helps accelerate genomic research, we look forward to its results - a quickened pace of scientific discovery and insight into biological processes and the causes of disease.&quot;</p>

<p>The GenePattern software package allows researchers to use a wide variety of methodologies to analyze gene expression data. It is also part of a larger architecture that addresses more general challenges in computational genomic research: the need to add new analysis tools quickly and easily, the need to reproduce the results of complex <i>in silico</i> analyses that require the coordination of different tools, and the need for a range of interfaces that make analyses accessible to non-programming researchers without limiting the full power of a tool to more experienced users.</p>

<p>GenePattern was recently selected by the Harvard Medical School-Partners Healthcare Center for Genetics and Genomics (HPCGG) for integration with their GIGPAD (Gateway for Integrated Genomics-Proteomics Application and Data) informatics system. &quot;GenePattern is the best solution for integrating bioinformatics processing capabilities into GIGPAD&#39;s research and clinical process flows. We&#39;ve been very pleased with GenePattern&#39;s functionality and well thought out implementation,&quot; said Samuel Aronson, director of information technology at the HPCGG. GIGPAD also received a 2005 <i>Bio-IT World</i> Best Practices award.</p>

<p>&quot;This is a well-deserved reward for the GenePattern team,&quot; said Jill Mesirov, principal investigator and director of Computational Biology and Bioinformatics at the Broad Institute. &quot;They made our vision for GenePattern - ease of use, interoperability, reproducibility, and flexibility - a reality.&quot;</p>

<p>Broad researchers recently released GenePattern 1.4, an enhanced version of the software package. For a comprehensive list of new features and fixes, and free access to the software, visit the GenePattern web site <a href="http://genepattern.org" target="_blank">http://www.genepattern.org</a>.</p>

<p>The 2005 <i>Bio-IT World</i> Best Practices celebration took place at the National Press Club in Washington, D.C. Submissions for the awards came from academic institutions, pharmaceutical and biotech companies.</p>

<p>Originally made available to the scientific community in 2004, there are currently 1,100 GenePattern users, including more than 500 institutions and 30 pharma-biotech companies across 35 countries worldwide.</p>

<p>In addition to Mesirov and Reich, members of the GenePattern development team include: Gad Getz, Josh Gould, Charlotte Henson, Jim Lerner, Ted Liefeld, Stefano Monti, Ken Ross, Pablo Tamayo, and Aravind Subramanian.</p>

<p>The work was supported by the National Institutes of Health.</p>
</div>
<!-- Article #4 -->

<div class="well well-lg">
<h3>Broad Institute researchers release GenePattern 1.4</h3>

<p><strong>By Broad Institute Communications<br />
Published June 17, 2005</strong></p>

<p><img class="press_image" height="89" src="http://www.broadinstitute.org/files/news/stories/full/genepatternicon.gif" width="90" /></p>

<p>Recently, researchers at the Broad Institute released GenePattern 1.4, an enhanced version of the microarray analysis package that is freely available to the scientific community.</p>

<p>GenePattern addresses several hurdles facing biomedical research, particularly the need for interoperable tools and reproducible <i>in silico</i> research in bioinformatics. It provides an accessible yet sophisticated analytical tool, with the ability to support novel approaches to analysis, while unifying investigative methods across disciplines.</p>

<p>Major new features of GenePattern 1.4 include:</p>

<ul>
	<li><b>Functionality for reproducible research </b>- Researchers can capture and package entire methodologies and export the information to colleagues for easy reproduction of analyses. This also provides researchers with the option of publishing <i>in silico</i> studies as a supplement to publications.</li>
</ul>

<ul>
	<li><b>File navigation aids </b>- Users can right-click on any file to view analysis options that can be run on a selected file.</li>
</ul>

<ul>
	<li><b>Industry standard compatibility</b> - Researchers can work in MAGE-ML format, as well as download data from the Gene Expression Omnibus database.</li>
</ul>

<p>The updated software and a comprehensive list of new features and fixes can be found on the GenePattern web site <a href="http://www.genepattern.org" target="_blank">http://www.genepattern.org</a>.</p>
</div>

<p>Currently, there are 1,100 GenePattern users, including more than 500 institutions and 30 pharma-biotech companies across 35 countries worldwide.</p>
<!-- Article #5 -->

<div class="well well-lg">
<h3>Broad Institute researchers release enhanced GenePattern software</h3>

<p><strong>By Broad Institute Communications<br />
Published September 23, 2004</strong></p>

<p>Recently, researchers at Broad Institute released GenePattern 1.2, an enhanced version of the GenePattern microarray analysis package that is freely available to the scientific community</p>

<p>The software offers an efficient, flexible microarray analysis tool that allows biomedical researchers across disciplines to perform custom gene expression analysis experiments, record and replay analyses, and use tools from many different software sources within a single interface. Major new features of GenePattern 1.2 include:</p>

<ul>
	<li><b>Online module repository</b>. A repository at the Broad Institute website allows users to identify, and select for automatic download, the modules they wish to import into the GenePattern system. This feature allows easy and automatic update of GenePattern as new modules are released.</li>
	<li><b>Java Programming Environment</b>. A programming library allows Java programmers to call any GenePattern module from within a Java application. Users can also save analysis pipelines as Java applications.</li>
	<li><b>Batch analyses</b>. A user can select a folder instead of a single file as input to a module, and the analysis will run on all of the files within that folder.</li>
</ul>

<p>The GenePattern 1.2 software and a comprehensive list of new features and fixes can be found on the <a href="http://www.genepattern.org" target="_blank">GenePattern website</a>.</p>

<p>Currently, over 200 institutions, including more than 30 pharma-biotech companies, across 35 countries worldwide, are using the software package.</p>

<p>GenePattern was developed with funding from the National Cancer Institute through a grant from the Biomedical Information Science and Technology Initiative of the National Institutes of Health. This grant was awarded to principal investigator Jill Mesirov, chief informatics officer and director of the Bioinformatics and Computational Biology Organization at Broad.</p>

<p>The Broad Institute (rhymes with &quot;code&quot;) is known officially as The Eli and Edythe L. Broad Institute. It is a research collaboration of the Massachusetts Institute of Technology, Harvard University and its hospitals and the Whitehead Institute for Biomedical Research. The Broad mission is to create comprehensive tools for genomic medicine, make them freely available to scientists worldwide and pioneer their application to understand and treat disease.</p>
</div>
</body>
</html>